{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Assemblies Quality Selection for NCBI Downloaded Assemblies"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "import matplotlib.pyplot as plt\n",
    "import xml.etree.ElementTree as ET\n",
    "import re"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## This script takes output list of biosamples from each species from the R script organize species"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "seq_path = \"/Users/rx32940/Dropbox/5.Rachel-projects/Phylogeography/Lepto_assemblies_V2/ncbi_assemblies\"\n",
    "\n",
    "# get isolates in each species\n",
    "species_table = pd.read_csv(seq_path + \"/species_list/Leptospira interrogans\", sep=\",\") \n",
    "species_acc = species_table['V1']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "biosamples = pd.DataFrame()\n",
    "\n",
    "biosamples = pd.read_csv(seq_path+\"/multiqc_quast.txt\", sep= \"\\t\") # read in output of multiqc\n",
    "biosamples['biosample_acc'] = biosamples[\"Sample\"].str.split(\" | \").str[0] # must split by \" | \", or biosample str will have a space in the str, can't be merged in later steps\n",
    "biosamples = biosamples.drop(\"Sample\",1)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## For NCBI downloaded assemblies coverage extraction "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Leptospira interrogans', 'Leptospira sp.',\n",
       "       'Leptospira borgpetersenii', 'Leptospira biflexa',\n",
       "       'Leptospira noguchii', 'Leptospira santarosai',\n",
       "       'Leptospira weilii', 'Leptospira licerasiae',\n",
       "       'Leptospira kirschneri', 'Leptospira inadai', 'Leptospira broomii',\n",
       "       'Leptospira kmetyi', 'Leptospira alexanderi', 'Leptospira meyeri',\n",
       "       'Leptospira wolffii', 'Leptospira fainei',\n",
       "       'Leptospira mayottensis', 'Leptospira sp. Fiocruz LV3954',\n",
       "       'Leptospira vanthielii', 'Leptospira yanagawae',\n",
       "       'Leptospira terpstrae', 'Leptospira wolbachii',\n",
       "       'Leptospira alstonii', 'Leptospira sp. Fiocruz LV4135',\n",
       "       'Leptospira sp. P2653', 'Leptospira sp. B5-022',\n",
       "       'Leptospira sp. serovar Kenya', 'Leptospira sp. ZV016',\n",
       "       'Leptospira tipperaryensis', 'Leptonema illini',\n",
       "       'Leptospira venezuelensis', 'Leptospira hartskeerlii',\n",
       "       'Leptospira saintgironsiae', 'Leptospira adleri',\n",
       "       'Leptospira perolatii', 'Leptospira barantonii',\n",
       "       'Leptospira harrisiae', 'Leptospira elllisii',\n",
       "       'Leptospira macculloughii', 'Leptospira haakeii',\n",
       "       'Leptospira levettii', 'Leptospira brenneri',\n",
       "       'Leptospira neocaledonica', 'Leptospira johnsonii',\n",
       "       'Leptospira ellinghausenii', 'Leptospira kobayashii',\n",
       "       'Leptospira ryugenii', 'Leptospira yasudae',\n",
       "       'Leptospira stimsonii', 'Leptospira fletcheri',\n",
       "       'Leptospira kanakyensis', 'Leptospira bourretii',\n",
       "       'Leptospira selangorensis', 'Leptospira gomenensis',\n",
       "       'Leptospira montravelensis', 'Leptospira bouyouniensis',\n",
       "       'Leptospira koniamboensis', 'Leptospira perdikensis',\n",
       "       'Leptospira langatensis', 'Leptospira sarikeiensis',\n",
       "       'Leptospira jelokensis', 'Leptospira kemamanensis',\n",
       "       'Leptospira dzianensis', 'Leptospira putramalaysiae',\n",
       "       'Leptospira semungkisensis', 'Leptospira andrefontaineae',\n",
       "       'Leptospira congkakensis', 'Leptospira mtsangambouensis',\n",
       "       'Leptospira bandrabouensis', 'Leptospira ognonensis',\n",
       "       'Leptospira noumeaensis', 'Leptospira dzoumogneensis',\n",
       "       'Leptospira idonii', 'Leptospira ilyithenensis',\n",
       "       'Leptospira fluminis'], dtype=object)"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# get coverage for each isolate\n",
    "path = \"/Users/rx32940/Dropbox/5.Rachel-projects/Phylogeography/Lepto_assemblies_V2/ncbi_assemblies/\"\n",
    "\n",
    "# the xml file was fixed by addind a root and heading (one xml can only have one root)\n",
    "doc = ET.parse(path + \"ncbi_assemblies_full.xml\") # xml object, biosample result file downloaded directly from ncbi with all_assemblies list\n",
    "\n",
    "root = doc.getroot() # get the root element\n",
    "\n",
    "full_cov = pd.DataFrame()\n",
    "for isolate in root.findall(\"DocumentSummary\"):\n",
    "    acc = isolate.find(\"BioSampleAccn\").text\n",
    "    cov = isolate.find(\"Coverage\").text\n",
    "    org = isolate.find(\"Organism\").text\n",
    "    species = isolate.find(\"SpeciesName\").text\n",
    "    N50 = isolate.find(\"ScaffoldN50\").text\n",
    "    inst = isolate.find(\"SubmitterOrganization\").text\n",
    "    complete = isolate.find(\"AssemblyStatus\").text\n",
    "    proj = isolate.find(\"GB_BioProjects\").find(\"Bioproj\").find(\"BioprojectAccn\").text\n",
    "    \n",
    "    coverage = pd.DataFrame({\"biosample_acc\":[acc],\"coverage\":[cov],\"organism\":[org],\"species\":[species], \"scaffoldN50\": [N50], \"Institue\":[inst], \"Assembly_status\":[complete],\"bioproject\":[proj]},\n",
    "                            columns = [\"biosample_acc\",\"coverage\",\"organism\",\"species\",\"scaffoldN50\",\"Institue\",\"Assembly_status\",\"bioproject\"])\n",
    "    full_cov = full_cov.append(coverage)\n",
    "    \n",
    "# ncbi downloaded assemblies and their species name\n",
    "species_dict = full_cov[[\"biosample_acc\",\"species\"]]\n",
    "\n",
    "species_dict.to_csv(path + \"ncbi_acc_species_dict.csv\",header=None, index=False)\n",
    "\n",
    "species_dict\n",
    "\n",
    "pd.unique(species_dict[\"species\"])\n",
    "    \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'SAMN06434252'"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "full_cov\n",
    "full_cov.to_csv(path+ \"assemblies_coverage.csv\") \n",
    "full_cov = pd.read_csv(path+\"assemblies_coverage.csv\").drop(\"Unnamed: 0\",1)\n",
    "full_cov[\"biosample_acc\"][0]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>biosample_acc</th>\n",
       "      <th>coverage</th>\n",
       "      <th>organism</th>\n",
       "      <th>species</th>\n",
       "      <th>scaffoldN50</th>\n",
       "      <th>Institue</th>\n",
       "      <th>Assembly_status</th>\n",
       "      <th>bioproject</th>\n",
       "      <th>L50</th>\n",
       "      <th>Duplication ratio</th>\n",
       "      <th>...</th>\n",
       "      <th>Misassembled contigs length</th>\n",
       "      <th>Reference length</th>\n",
       "      <th># misassembled contigs</th>\n",
       "      <th>NGA75</th>\n",
       "      <th>NA75</th>\n",
       "      <th>LGA75</th>\n",
       "      <th>NA50</th>\n",
       "      <th>LGA50</th>\n",
       "      <th>LA75</th>\n",
       "      <th>LA50</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SAMN06434252</td>\n",
       "      <td>30.0000</td>\n",
       "      <td>Leptospira interrogans serovar Canicola (bacte...</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>4218946</td>\n",
       "      <td>Universidade Federal de Pelotas</td>\n",
       "      <td>Chromosome</td>\n",
       "      <td>PRJNA376496</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.003</td>\n",
       "      <td>...</td>\n",
       "      <td>4570695.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>30352.0</td>\n",
       "      <td>32897.0</td>\n",
       "      <td>52.0</td>\n",
       "      <td>62085.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>49.0</td>\n",
       "      <td>23.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SAMN06434273</td>\n",
       "      <td>30.0000</td>\n",
       "      <td>Leptospira interrogans serovar Canicola (bacte...</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>4208129</td>\n",
       "      <td>Universidade Federal de Pelotas</td>\n",
       "      <td>Chromosome</td>\n",
       "      <td>PRJNA376501</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.007</td>\n",
       "      <td>...</td>\n",
       "      <td>4560160.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>28968.0</td>\n",
       "      <td>30585.0</td>\n",
       "      <td>54.0</td>\n",
       "      <td>60422.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>51.0</td>\n",
       "      <td>23.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SAMN07572185</td>\n",
       "      <td>15.4200</td>\n",
       "      <td>Leptospira sp. (bacteria)</td>\n",
       "      <td>Leptospira sp.</td>\n",
       "      <td>65100</td>\n",
       "      <td>University of Illinois at Urbana-Champaign</td>\n",
       "      <td>Contig</td>\n",
       "      <td>PRJNA323575</td>\n",
       "      <td>15.0</td>\n",
       "      <td>1.000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SAMN11389100</td>\n",
       "      <td>20.0000</td>\n",
       "      <td>Leptospira borgpetersenii (bacteria)</td>\n",
       "      <td>Leptospira borgpetersenii</td>\n",
       "      <td>3580756</td>\n",
       "      <td>INDICASAT AIP</td>\n",
       "      <td>Chromosome</td>\n",
       "      <td>PRJNA543109</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.004</td>\n",
       "      <td>...</td>\n",
       "      <td>3887834.0</td>\n",
       "      <td>3931782.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>15022.0</td>\n",
       "      <td>15795.0</td>\n",
       "      <td>80.0</td>\n",
       "      <td>34042.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>78.0</td>\n",
       "      <td>37.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SAMN11389098</td>\n",
       "      <td>20.0000</td>\n",
       "      <td>Leptospira borgpetersenii (bacteria)</td>\n",
       "      <td>Leptospira borgpetersenii</td>\n",
       "      <td>3579904</td>\n",
       "      <td>INDICASAT AIP</td>\n",
       "      <td>Chromosome</td>\n",
       "      <td>PRJNA543109</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.004</td>\n",
       "      <td>...</td>\n",
       "      <td>3887110.0</td>\n",
       "      <td>3931782.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>14988.0</td>\n",
       "      <td>15170.0</td>\n",
       "      <td>81.0</td>\n",
       "      <td>34043.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>79.0</td>\n",
       "      <td>37.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>SAMN11389099</td>\n",
       "      <td>20.0000</td>\n",
       "      <td>Leptospira borgpetersenii (bacteria)</td>\n",
       "      <td>Leptospira borgpetersenii</td>\n",
       "      <td>3580481</td>\n",
       "      <td>INDICASAT AIP</td>\n",
       "      <td>Chromosome</td>\n",
       "      <td>PRJNA543109</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.004</td>\n",
       "      <td>...</td>\n",
       "      <td>3887691.0</td>\n",
       "      <td>3931782.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>15022.0</td>\n",
       "      <td>15795.0</td>\n",
       "      <td>80.0</td>\n",
       "      <td>34043.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>78.0</td>\n",
       "      <td>37.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>SAMN11389084</td>\n",
       "      <td>20.0000</td>\n",
       "      <td>Leptospira borgpetersenii (bacteria)</td>\n",
       "      <td>Leptospira borgpetersenii</td>\n",
       "      <td>3513910</td>\n",
       "      <td>INDICASAT AIP</td>\n",
       "      <td>Chromosome</td>\n",
       "      <td>PRJNA543109</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.004</td>\n",
       "      <td>...</td>\n",
       "      <td>3821116.0</td>\n",
       "      <td>3931782.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>14571.0</td>\n",
       "      <td>15170.0</td>\n",
       "      <td>83.0</td>\n",
       "      <td>34725.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>77.0</td>\n",
       "      <td>36.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>SAMN11389101</td>\n",
       "      <td>20.0000</td>\n",
       "      <td>Leptospira borgpetersenii (bacteria)</td>\n",
       "      <td>Leptospira borgpetersenii</td>\n",
       "      <td>3581767</td>\n",
       "      <td>INDICASAT AIP</td>\n",
       "      <td>Chromosome</td>\n",
       "      <td>PRJNA543109</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.004</td>\n",
       "      <td>...</td>\n",
       "      <td>3888971.0</td>\n",
       "      <td>3931782.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>14988.0</td>\n",
       "      <td>15170.0</td>\n",
       "      <td>81.0</td>\n",
       "      <td>34043.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>79.0</td>\n",
       "      <td>37.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>SAMN11389090</td>\n",
       "      <td>20.0000</td>\n",
       "      <td>Leptospira interrogans (bacteria)</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>4233607</td>\n",
       "      <td>INDICASAT AIP</td>\n",
       "      <td>Chromosome</td>\n",
       "      <td>PRJNA543109</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.007</td>\n",
       "      <td>...</td>\n",
       "      <td>4583691.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>60938.0</td>\n",
       "      <td>61586.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>132556.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>SAMN11389094</td>\n",
       "      <td>20.0000</td>\n",
       "      <td>Leptospira interrogans (bacteria)</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>4213733</td>\n",
       "      <td>INDICASAT AIP</td>\n",
       "      <td>Chromosome</td>\n",
       "      <td>PRJNA543109</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.003</td>\n",
       "      <td>...</td>\n",
       "      <td>4563643.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>60640.0</td>\n",
       "      <td>61945.0</td>\n",
       "      <td>30.0</td>\n",
       "      <td>99422.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>13.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>SAMN11389096</td>\n",
       "      <td>20.0000</td>\n",
       "      <td>Leptospira interrogans (bacteria)</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>4214555</td>\n",
       "      <td>INDICASAT AIP</td>\n",
       "      <td>Chromosome</td>\n",
       "      <td>PRJNA543109</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.003</td>\n",
       "      <td>...</td>\n",
       "      <td>4564465.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>61945.0</td>\n",
       "      <td>64756.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>122749.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>12.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>SAMN11389097</td>\n",
       "      <td>20.0000</td>\n",
       "      <td>Leptospira interrogans (bacteria)</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>4206042</td>\n",
       "      <td>INDICASAT AIP</td>\n",
       "      <td>Chromosome</td>\n",
       "      <td>PRJNA543109</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.003</td>\n",
       "      <td>...</td>\n",
       "      <td>4556294.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>61630.0</td>\n",
       "      <td>64047.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>114234.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>13.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>SAMN11389093</td>\n",
       "      <td>20.0000</td>\n",
       "      <td>Leptospira interrogans (bacteria)</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>4213648</td>\n",
       "      <td>INDICASAT AIP</td>\n",
       "      <td>Chromosome</td>\n",
       "      <td>PRJNA543109</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.003</td>\n",
       "      <td>...</td>\n",
       "      <td>4563558.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>61630.0</td>\n",
       "      <td>64047.0</td>\n",
       "      <td>30.0</td>\n",
       "      <td>110881.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>14.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>SAMN11389089</td>\n",
       "      <td>20.0000</td>\n",
       "      <td>Leptospira interrogans (bacteria)</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>4214175</td>\n",
       "      <td>INDICASAT AIP</td>\n",
       "      <td>Chromosome</td>\n",
       "      <td>PRJNA543109</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.003</td>\n",
       "      <td>...</td>\n",
       "      <td>4564085.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>61630.0</td>\n",
       "      <td>63947.0</td>\n",
       "      <td>30.0</td>\n",
       "      <td>110881.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>14.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>SAMN11389091</td>\n",
       "      <td>20.0000</td>\n",
       "      <td>Leptospira borgpetersenii (bacteria)</td>\n",
       "      <td>Leptospira borgpetersenii</td>\n",
       "      <td>3601002</td>\n",
       "      <td>INDICASAT AIP</td>\n",
       "      <td>Chromosome</td>\n",
       "      <td>PRJNA543109</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.005</td>\n",
       "      <td>...</td>\n",
       "      <td>3908796.0</td>\n",
       "      <td>3931782.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>15022.0</td>\n",
       "      <td>15169.0</td>\n",
       "      <td>82.0</td>\n",
       "      <td>34043.0</td>\n",
       "      <td>38.0</td>\n",
       "      <td>81.0</td>\n",
       "      <td>37.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>SAMN11389087</td>\n",
       "      <td>20.0000</td>\n",
       "      <td>Leptospira interrogans (bacteria)</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>4214328</td>\n",
       "      <td>INDICASAT AIP</td>\n",
       "      <td>Chromosome</td>\n",
       "      <td>PRJNA543109</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.002</td>\n",
       "      <td>...</td>\n",
       "      <td>4564238.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>61630.0</td>\n",
       "      <td>63947.0</td>\n",
       "      <td>30.0</td>\n",
       "      <td>110881.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>14.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>SAMN11389086</td>\n",
       "      <td>20.0000</td>\n",
       "      <td>Leptospira interrogans (bacteria)</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>4212939</td>\n",
       "      <td>INDICASAT AIP</td>\n",
       "      <td>Chromosome</td>\n",
       "      <td>PRJNA543109</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.003</td>\n",
       "      <td>...</td>\n",
       "      <td>4563191.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>61630.0</td>\n",
       "      <td>64047.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>110881.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>14.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>SAMN11389085</td>\n",
       "      <td>20.0000</td>\n",
       "      <td>Leptospira borgpetersenii (bacteria)</td>\n",
       "      <td>Leptospira borgpetersenii</td>\n",
       "      <td>3580531</td>\n",
       "      <td>INDICASAT AIP</td>\n",
       "      <td>Chromosome</td>\n",
       "      <td>PRJNA543109</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.004</td>\n",
       "      <td>...</td>\n",
       "      <td>3887609.0</td>\n",
       "      <td>3931782.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>14988.0</td>\n",
       "      <td>15795.0</td>\n",
       "      <td>81.0</td>\n",
       "      <td>34043.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>78.0</td>\n",
       "      <td>37.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>SAMN11389092</td>\n",
       "      <td>20.0000</td>\n",
       "      <td>Leptospira borgpetersenii (bacteria)</td>\n",
       "      <td>Leptospira borgpetersenii</td>\n",
       "      <td>3581013</td>\n",
       "      <td>INDICASAT AIP</td>\n",
       "      <td>Chromosome</td>\n",
       "      <td>PRJNA543109</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.004</td>\n",
       "      <td>...</td>\n",
       "      <td>3888127.0</td>\n",
       "      <td>3931782.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>15022.0</td>\n",
       "      <td>15883.0</td>\n",
       "      <td>79.0</td>\n",
       "      <td>34725.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>77.0</td>\n",
       "      <td>36.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>SAMN11389095</td>\n",
       "      <td>20.0000</td>\n",
       "      <td>Leptospira interrogans (bacteria)</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>4213431</td>\n",
       "      <td>INDICASAT AIP</td>\n",
       "      <td>Chromosome</td>\n",
       "      <td>PRJNA543109</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.003</td>\n",
       "      <td>...</td>\n",
       "      <td>4563683.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>61630.0</td>\n",
       "      <td>63947.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>114234.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>13.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>SAMN11389088</td>\n",
       "      <td>20.0000</td>\n",
       "      <td>Leptospira interrogans (bacteria)</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>4223477</td>\n",
       "      <td>INDICASAT AIP</td>\n",
       "      <td>Chromosome</td>\n",
       "      <td>PRJNA543109</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.003</td>\n",
       "      <td>...</td>\n",
       "      <td>4573647.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>63194.0</td>\n",
       "      <td>66836.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>115598.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>12.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>SAMN02603847</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>Leptospira interrogans serovar Copenhageni str...</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>4277185</td>\n",
       "      <td>Sao Paulo state (Brazil) Consortium</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>PRJNA10687</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.004</td>\n",
       "      <td>...</td>\n",
       "      <td>4627366.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>46921.0</td>\n",
       "      <td>47908.0</td>\n",
       "      <td>38.0</td>\n",
       "      <td>80954.0</td>\n",
       "      <td>19.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>19.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>SAMN02603615</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>Leptospira borgpetersenii serovar Hardjo-bovis...</td>\n",
       "      <td>Leptospira borgpetersenii</td>\n",
       "      <td>3614446</td>\n",
       "      <td>Monash University</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>PRJNA16146</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.073</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>SAMN02603616</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>Leptospira borgpetersenii serovar Hardjo-bovis...</td>\n",
       "      <td>Leptospira borgpetersenii</td>\n",
       "      <td>3576473</td>\n",
       "      <td>Monash University</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>PRJNA16148</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.052</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>SAMN02603617</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>Leptospira biflexa serovar Patoc strain 'Patoc...</td>\n",
       "      <td>Leptospira biflexa</td>\n",
       "      <td>3603977</td>\n",
       "      <td>Monash University</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>PRJNA16153</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>SAMN02603347</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>Leptospira biflexa serovar Patoc strain 'Patoc...</td>\n",
       "      <td>Leptospira biflexa</td>\n",
       "      <td>3599677</td>\n",
       "      <td>Institut Pasteur</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>PRJNA20133</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>SAMN02603127</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>Leptospira interrogans serovar Lai str. 56601 ...</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>4338762</td>\n",
       "      <td>Chinese National HGC, Shanghai</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>PRJNA293</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4338762.0</td>\n",
       "      <td>4338762.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4338762.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>SAMN00254366</td>\n",
       "      <td>19.3900</td>\n",
       "      <td>Leptospira interrogans serovar Copenhageni str...</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>19190</td>\n",
       "      <td>J. Craig Venter Institute</td>\n",
       "      <td>Scaffold</td>\n",
       "      <td>PRJNA65081</td>\n",
       "      <td>74.0</td>\n",
       "      <td>1.005</td>\n",
       "      <td>...</td>\n",
       "      <td>943347.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>50.0</td>\n",
       "      <td>7346.0</td>\n",
       "      <td>8395.0</td>\n",
       "      <td>198.0</td>\n",
       "      <td>16502.0</td>\n",
       "      <td>91.0</td>\n",
       "      <td>182.0</td>\n",
       "      <td>86.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>SAMN00254327</td>\n",
       "      <td>25.5500</td>\n",
       "      <td>Leptospira interrogans serovar Copenhageni str...</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>27585</td>\n",
       "      <td>J. Craig Venter Institute</td>\n",
       "      <td>Scaffold</td>\n",
       "      <td>PRJNA65079</td>\n",
       "      <td>49.0</td>\n",
       "      <td>1.004</td>\n",
       "      <td>...</td>\n",
       "      <td>1147421.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>41.0</td>\n",
       "      <td>12247.0</td>\n",
       "      <td>13704.0</td>\n",
       "      <td>128.0</td>\n",
       "      <td>23020.0</td>\n",
       "      <td>60.0</td>\n",
       "      <td>119.0</td>\n",
       "      <td>56.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>SAMN00254376</td>\n",
       "      <td>19.5500</td>\n",
       "      <td>Leptospira interrogans serovar Copenhageni str...</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>29809</td>\n",
       "      <td>J. Craig Venter Institute</td>\n",
       "      <td>Scaffold</td>\n",
       "      <td>PRJNA65077</td>\n",
       "      <td>50.0</td>\n",
       "      <td>1.004</td>\n",
       "      <td>...</td>\n",
       "      <td>1482896.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>54.0</td>\n",
       "      <td>11357.0</td>\n",
       "      <td>12984.0</td>\n",
       "      <td>132.0</td>\n",
       "      <td>26118.0</td>\n",
       "      <td>61.0</td>\n",
       "      <td>121.0</td>\n",
       "      <td>57.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>621</th>\n",
       "      <td>SAMN10316368</td>\n",
       "      <td>50.0000</td>\n",
       "      <td>Leptospira dzoumogneensis (bacteria)</td>\n",
       "      <td>Leptospira dzoumogneensis</td>\n",
       "      <td>585855</td>\n",
       "      <td>INRS-Institut Armand-Frappier</td>\n",
       "      <td>Contig</td>\n",
       "      <td>PRJNA480513</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.135</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>622</th>\n",
       "      <td>SAMN10316367</td>\n",
       "      <td>50.0000</td>\n",
       "      <td>Leptospira bandrabouensis (bacteria)</td>\n",
       "      <td>Leptospira bandrabouensis</td>\n",
       "      <td>2180649</td>\n",
       "      <td>INRS-Institut Armand-Frappier</td>\n",
       "      <td>Contig</td>\n",
       "      <td>PRJNA480513</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>623</th>\n",
       "      <td>SAMN10316389</td>\n",
       "      <td>50.0000</td>\n",
       "      <td>Leptospira levettii (bacteria)</td>\n",
       "      <td>Leptospira levettii</td>\n",
       "      <td>884966</td>\n",
       "      <td>INRS-Institut Armand-Frappier</td>\n",
       "      <td>Contig</td>\n",
       "      <td>PRJNA480513</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.635</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>624</th>\n",
       "      <td>SAMN10316379</td>\n",
       "      <td>50.0000</td>\n",
       "      <td>Leptospira meyeri (bacteria)</td>\n",
       "      <td>Leptospira meyeri</td>\n",
       "      <td>1004763</td>\n",
       "      <td>INRS-Institut Armand-Frappier</td>\n",
       "      <td>Contig</td>\n",
       "      <td>PRJNA480513</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>625</th>\n",
       "      <td>SAMN10316374</td>\n",
       "      <td>50.0000</td>\n",
       "      <td>Leptospira levettii (bacteria)</td>\n",
       "      <td>Leptospira levettii</td>\n",
       "      <td>2751354</td>\n",
       "      <td>INRS-Institut Armand-Frappier</td>\n",
       "      <td>Contig</td>\n",
       "      <td>PRJNA480513</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.776</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>626</th>\n",
       "      <td>SAMN10316364</td>\n",
       "      <td>50.0000</td>\n",
       "      <td>Leptospira idonii (bacteria)</td>\n",
       "      <td>Leptospira idonii</td>\n",
       "      <td>195150</td>\n",
       "      <td>INRS-Institut Armand-Frappier</td>\n",
       "      <td>Contig</td>\n",
       "      <td>PRJNA480513</td>\n",
       "      <td>7.0</td>\n",
       "      <td>1.000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>627</th>\n",
       "      <td>SAMN10316365</td>\n",
       "      <td>50.0000</td>\n",
       "      <td>Leptospira ilyithenensis (bacteria)</td>\n",
       "      <td>Leptospira ilyithenensis</td>\n",
       "      <td>266913</td>\n",
       "      <td>INRS-Institut Armand-Frappier</td>\n",
       "      <td>Contig</td>\n",
       "      <td>PRJNA480513</td>\n",
       "      <td>5.0</td>\n",
       "      <td>1.000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>628</th>\n",
       "      <td>SAMN10316373</td>\n",
       "      <td>50.0000</td>\n",
       "      <td>Leptospira levettii (bacteria)</td>\n",
       "      <td>Leptospira levettii</td>\n",
       "      <td>2854332</td>\n",
       "      <td>INRS-Institut Armand-Frappier</td>\n",
       "      <td>Contig</td>\n",
       "      <td>PRJNA480513</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.239</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>629</th>\n",
       "      <td>SAMN10316443</td>\n",
       "      <td>50.0000</td>\n",
       "      <td>Leptospira fluminis (bacteria)</td>\n",
       "      <td>Leptospira fluminis</td>\n",
       "      <td>325090</td>\n",
       "      <td>INRS-Institut Armand-Frappier</td>\n",
       "      <td>Contig</td>\n",
       "      <td>PRJNA480513</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.146</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>630</th>\n",
       "      <td>SAMN11289419</td>\n",
       "      <td>30.0000</td>\n",
       "      <td>Leptospira interrogans (bacteria)</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>4306144</td>\n",
       "      <td>Rajarata University of Sri lanka</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>PRJNA528695</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.010</td>\n",
       "      <td>...</td>\n",
       "      <td>4661889.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>28022.0</td>\n",
       "      <td>26733.0</td>\n",
       "      <td>56.0</td>\n",
       "      <td>52332.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>58.0</td>\n",
       "      <td>27.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>631</th>\n",
       "      <td>SAMN11289420</td>\n",
       "      <td>30.0000</td>\n",
       "      <td>Leptospira interrogans (bacteria)</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>4293951</td>\n",
       "      <td>Rajarata University of Sri lanka</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>PRJNA528695</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.004</td>\n",
       "      <td>...</td>\n",
       "      <td>4651137.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>33799.0</td>\n",
       "      <td>34493.0</td>\n",
       "      <td>54.0</td>\n",
       "      <td>54269.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>52.0</td>\n",
       "      <td>27.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>632</th>\n",
       "      <td>SAMN11289421</td>\n",
       "      <td>30.0000</td>\n",
       "      <td>Leptospira interrogans (bacteria)</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>4497709</td>\n",
       "      <td>Rajarata University of Sri lanka</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>PRJNA528695</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.012</td>\n",
       "      <td>...</td>\n",
       "      <td>4858092.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>31632.0</td>\n",
       "      <td>20236.0</td>\n",
       "      <td>50.0</td>\n",
       "      <td>53490.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>62.0</td>\n",
       "      <td>27.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>633</th>\n",
       "      <td>SAMN11774880</td>\n",
       "      <td>101.0000</td>\n",
       "      <td>Leptospira noguchii (bacteria)</td>\n",
       "      <td>Leptospira noguchii</td>\n",
       "      <td>15273</td>\n",
       "      <td>INRS - Institut Armand-Frappier</td>\n",
       "      <td>Contig</td>\n",
       "      <td>PRJNA543681</td>\n",
       "      <td>95.0</td>\n",
       "      <td>1.235</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>634</th>\n",
       "      <td>SAMN11774879</td>\n",
       "      <td>95.0000</td>\n",
       "      <td>Leptospira borgpetersenii (bacteria)</td>\n",
       "      <td>Leptospira borgpetersenii</td>\n",
       "      <td>35480</td>\n",
       "      <td>INRS - Institut Armand-Frappier</td>\n",
       "      <td>Contig</td>\n",
       "      <td>PRJNA543681</td>\n",
       "      <td>30.0</td>\n",
       "      <td>1.339</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>635</th>\n",
       "      <td>SAMN11774878</td>\n",
       "      <td>96.0000</td>\n",
       "      <td>Leptospira borgpetersenii (bacteria)</td>\n",
       "      <td>Leptospira borgpetersenii</td>\n",
       "      <td>34743</td>\n",
       "      <td>INRS - Institut Armand-Frappier</td>\n",
       "      <td>Contig</td>\n",
       "      <td>PRJNA543681</td>\n",
       "      <td>28.0</td>\n",
       "      <td>1.316</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>636</th>\n",
       "      <td>SAMN11774876</td>\n",
       "      <td>98.0000</td>\n",
       "      <td>Leptospira interrogans (bacteria)</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>41872</td>\n",
       "      <td>INRS - Institut Armand-Frappier</td>\n",
       "      <td>Contig</td>\n",
       "      <td>PRJNA543681</td>\n",
       "      <td>31.0</td>\n",
       "      <td>1.007</td>\n",
       "      <td>...</td>\n",
       "      <td>2217479.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>59.0</td>\n",
       "      <td>12932.0</td>\n",
       "      <td>14831.0</td>\n",
       "      <td>109.0</td>\n",
       "      <td>29836.0</td>\n",
       "      <td>47.0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>44.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>637</th>\n",
       "      <td>SAMN11774877</td>\n",
       "      <td>98.0000</td>\n",
       "      <td>Leptospira interrogans (bacteria)</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>49707</td>\n",
       "      <td>INRS - Institut Armand-Frappier</td>\n",
       "      <td>Contig</td>\n",
       "      <td>PRJNA543681</td>\n",
       "      <td>29.0</td>\n",
       "      <td>1.007</td>\n",
       "      <td>...</td>\n",
       "      <td>2050612.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>52.0</td>\n",
       "      <td>13590.0</td>\n",
       "      <td>14976.0</td>\n",
       "      <td>106.0</td>\n",
       "      <td>31296.0</td>\n",
       "      <td>45.0</td>\n",
       "      <td>97.0</td>\n",
       "      <td>42.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>638</th>\n",
       "      <td>SAMN11774874</td>\n",
       "      <td>98.0000</td>\n",
       "      <td>Leptospira interrogans (bacteria)</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>38918</td>\n",
       "      <td>INRS - Institut Armand-Frappier</td>\n",
       "      <td>Contig</td>\n",
       "      <td>PRJNA543681</td>\n",
       "      <td>38.0</td>\n",
       "      <td>1.008</td>\n",
       "      <td>...</td>\n",
       "      <td>2055025.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>64.0</td>\n",
       "      <td>12336.0</td>\n",
       "      <td>14091.0</td>\n",
       "      <td>119.0</td>\n",
       "      <td>27872.0</td>\n",
       "      <td>53.0</td>\n",
       "      <td>109.0</td>\n",
       "      <td>50.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>639</th>\n",
       "      <td>SAMN11774875</td>\n",
       "      <td>98.0000</td>\n",
       "      <td>Leptospira interrogans (bacteria)</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>41894</td>\n",
       "      <td>INRS - Institut Armand-Frappier</td>\n",
       "      <td>Contig</td>\n",
       "      <td>PRJNA543681</td>\n",
       "      <td>31.0</td>\n",
       "      <td>1.007</td>\n",
       "      <td>...</td>\n",
       "      <td>2265076.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>58.0</td>\n",
       "      <td>12932.0</td>\n",
       "      <td>14191.0</td>\n",
       "      <td>109.0</td>\n",
       "      <td>29203.0</td>\n",
       "      <td>46.0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>43.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>640</th>\n",
       "      <td>SAMN11936725</td>\n",
       "      <td>38.0000</td>\n",
       "      <td>Leptospira weilii (bacteria)</td>\n",
       "      <td>Leptospira weilii</td>\n",
       "      <td>3978120</td>\n",
       "      <td>Chulalongkorn University</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>PRJNA545909</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.370</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>641</th>\n",
       "      <td>SAMN11936708</td>\n",
       "      <td>30.0000</td>\n",
       "      <td>Leptospira weilii (bacteria)</td>\n",
       "      <td>Leptospira weilii</td>\n",
       "      <td>3965053</td>\n",
       "      <td>Chulalongkorn University</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>PRJNA545906</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.370</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>642</th>\n",
       "      <td>SAMN12588099</td>\n",
       "      <td>215.0000</td>\n",
       "      <td>Leptospira interrogans serovar Hardjo (bacteria)</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>4355568</td>\n",
       "      <td>Londrina State University</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>PRJNA560522</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.022</td>\n",
       "      <td>...</td>\n",
       "      <td>4705503.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>35465.0</td>\n",
       "      <td>35198.0</td>\n",
       "      <td>47.0</td>\n",
       "      <td>66852.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>48.0</td>\n",
       "      <td>23.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>643</th>\n",
       "      <td>SAMN08513768</td>\n",
       "      <td>31.5296</td>\n",
       "      <td>Leptospira interrogans serovar Geyaweera (bact...</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>3284430</td>\n",
       "      <td>JCVI</td>\n",
       "      <td>Contig</td>\n",
       "      <td>PRJNA401006</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.017</td>\n",
       "      <td>...</td>\n",
       "      <td>4695993.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>29455.0</td>\n",
       "      <td>16143.0</td>\n",
       "      <td>59.0</td>\n",
       "      <td>46841.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>80.0</td>\n",
       "      <td>35.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>644</th>\n",
       "      <td>SAMN08596279</td>\n",
       "      <td>498.6260</td>\n",
       "      <td>Leptospira interrogans serovar Weerasinghe (ba...</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>4266484</td>\n",
       "      <td>JCVI</td>\n",
       "      <td>Contig</td>\n",
       "      <td>PRJNA401006</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.053</td>\n",
       "      <td>...</td>\n",
       "      <td>4947504.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>29375.0</td>\n",
       "      <td>22847.0</td>\n",
       "      <td>62.0</td>\n",
       "      <td>45717.0</td>\n",
       "      <td>31.0</td>\n",
       "      <td>72.0</td>\n",
       "      <td>35.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>645</th>\n",
       "      <td>SAMN12567456</td>\n",
       "      <td>888.0000</td>\n",
       "      <td>Leptospira interrogans serovar Copenhageni (ba...</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>83218</td>\n",
       "      <td>Oswaldo Cruz Foundation</td>\n",
       "      <td>Scaffold</td>\n",
       "      <td>PRJNA560059</td>\n",
       "      <td>19.0</td>\n",
       "      <td>1.003</td>\n",
       "      <td>...</td>\n",
       "      <td>2853158.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>41.0</td>\n",
       "      <td>23488.0</td>\n",
       "      <td>27692.0</td>\n",
       "      <td>66.0</td>\n",
       "      <td>48492.0</td>\n",
       "      <td>31.0</td>\n",
       "      <td>62.0</td>\n",
       "      <td>29.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>646</th>\n",
       "      <td>SAMN12871420</td>\n",
       "      <td>100.0000</td>\n",
       "      <td>Leptospira interrogans serovar Canicola (bacte...</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>4259066</td>\n",
       "      <td>Chinese National Human Genome Center at Shanghai</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>PRJNA574874</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.008</td>\n",
       "      <td>...</td>\n",
       "      <td>4616103.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>27538.0</td>\n",
       "      <td>26688.0</td>\n",
       "      <td>58.0</td>\n",
       "      <td>50605.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>60.0</td>\n",
       "      <td>30.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>647</th>\n",
       "      <td>SAMN12871448</td>\n",
       "      <td>100.0000</td>\n",
       "      <td>Leptospira interrogans serovar Canicola (bacte...</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>4255595</td>\n",
       "      <td>Chinese National Human Genome Center at Shanghai</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>PRJNA574874</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.008</td>\n",
       "      <td>...</td>\n",
       "      <td>4612880.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>27538.0</td>\n",
       "      <td>26688.0</td>\n",
       "      <td>59.0</td>\n",
       "      <td>50648.0</td>\n",
       "      <td>30.0</td>\n",
       "      <td>61.0</td>\n",
       "      <td>30.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>648</th>\n",
       "      <td>SAMN12567526</td>\n",
       "      <td>152.0000</td>\n",
       "      <td>Leptospira interrogans serovar Copenhageni (ba...</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>89408</td>\n",
       "      <td>Oswaldo Cruz Foundation</td>\n",
       "      <td>Scaffold</td>\n",
       "      <td>PRJNA560059</td>\n",
       "      <td>18.0</td>\n",
       "      <td>1.003</td>\n",
       "      <td>...</td>\n",
       "      <td>3040784.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>38.0</td>\n",
       "      <td>25000.0</td>\n",
       "      <td>26527.0</td>\n",
       "      <td>66.0</td>\n",
       "      <td>48083.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>63.0</td>\n",
       "      <td>30.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>649</th>\n",
       "      <td>SAMN05421689</td>\n",
       "      <td>317.0000</td>\n",
       "      <td>Leptospira interrogans (bacteria)</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>133191</td>\n",
       "      <td>DOE - JOINT GENOME INSTITUTE</td>\n",
       "      <td>Scaffold</td>\n",
       "      <td>PRJEB18875</td>\n",
       "      <td>10.0</td>\n",
       "      <td>1.005</td>\n",
       "      <td>...</td>\n",
       "      <td>3949662.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>35959.0</td>\n",
       "      <td>37810.0</td>\n",
       "      <td>50.0</td>\n",
       "      <td>57457.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>48.0</td>\n",
       "      <td>23.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>650</th>\n",
       "      <td>SAMEA104369440</td>\n",
       "      <td>300.0000</td>\n",
       "      <td>Leptospira interrogans serovar Manilae (bacteria)</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>4614703</td>\n",
       "      <td>INSTITUT PASTEUR</td>\n",
       "      <td>Chromosome</td>\n",
       "      <td>PRJEB23342</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.003</td>\n",
       "      <td>...</td>\n",
       "      <td>4614703.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>22946.0</td>\n",
       "      <td>23335.0</td>\n",
       "      <td>77.0</td>\n",
       "      <td>38033.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>75.0</td>\n",
       "      <td>36.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>651 rows × 61 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      biosample_acc  coverage  \\\n",
       "0      SAMN06434252   30.0000   \n",
       "1      SAMN06434273   30.0000   \n",
       "2      SAMN07572185   15.4200   \n",
       "3      SAMN11389100   20.0000   \n",
       "4      SAMN11389098   20.0000   \n",
       "5      SAMN11389099   20.0000   \n",
       "6      SAMN11389084   20.0000   \n",
       "7      SAMN11389101   20.0000   \n",
       "8      SAMN11389090   20.0000   \n",
       "9      SAMN11389094   20.0000   \n",
       "10     SAMN11389096   20.0000   \n",
       "11     SAMN11389097   20.0000   \n",
       "12     SAMN11389093   20.0000   \n",
       "13     SAMN11389089   20.0000   \n",
       "14     SAMN11389091   20.0000   \n",
       "15     SAMN11389087   20.0000   \n",
       "16     SAMN11389086   20.0000   \n",
       "17     SAMN11389085   20.0000   \n",
       "18     SAMN11389092   20.0000   \n",
       "19     SAMN11389095   20.0000   \n",
       "20     SAMN11389088   20.0000   \n",
       "21     SAMN02603847    0.0000   \n",
       "22     SAMN02603615    0.0000   \n",
       "23     SAMN02603616    0.0000   \n",
       "24     SAMN02603617    0.0000   \n",
       "25     SAMN02603347    0.0000   \n",
       "26     SAMN02603127    0.0000   \n",
       "27     SAMN00254366   19.3900   \n",
       "28     SAMN00254327   25.5500   \n",
       "29     SAMN00254376   19.5500   \n",
       "..              ...       ...   \n",
       "621    SAMN10316368   50.0000   \n",
       "622    SAMN10316367   50.0000   \n",
       "623    SAMN10316389   50.0000   \n",
       "624    SAMN10316379   50.0000   \n",
       "625    SAMN10316374   50.0000   \n",
       "626    SAMN10316364   50.0000   \n",
       "627    SAMN10316365   50.0000   \n",
       "628    SAMN10316373   50.0000   \n",
       "629    SAMN10316443   50.0000   \n",
       "630    SAMN11289419   30.0000   \n",
       "631    SAMN11289420   30.0000   \n",
       "632    SAMN11289421   30.0000   \n",
       "633    SAMN11774880  101.0000   \n",
       "634    SAMN11774879   95.0000   \n",
       "635    SAMN11774878   96.0000   \n",
       "636    SAMN11774876   98.0000   \n",
       "637    SAMN11774877   98.0000   \n",
       "638    SAMN11774874   98.0000   \n",
       "639    SAMN11774875   98.0000   \n",
       "640    SAMN11936725   38.0000   \n",
       "641    SAMN11936708   30.0000   \n",
       "642    SAMN12588099  215.0000   \n",
       "643    SAMN08513768   31.5296   \n",
       "644    SAMN08596279  498.6260   \n",
       "645    SAMN12567456  888.0000   \n",
       "646    SAMN12871420  100.0000   \n",
       "647    SAMN12871448  100.0000   \n",
       "648    SAMN12567526  152.0000   \n",
       "649    SAMN05421689  317.0000   \n",
       "650  SAMEA104369440  300.0000   \n",
       "\n",
       "                                              organism  \\\n",
       "0    Leptospira interrogans serovar Canicola (bacte...   \n",
       "1    Leptospira interrogans serovar Canicola (bacte...   \n",
       "2                            Leptospira sp. (bacteria)   \n",
       "3                 Leptospira borgpetersenii (bacteria)   \n",
       "4                 Leptospira borgpetersenii (bacteria)   \n",
       "5                 Leptospira borgpetersenii (bacteria)   \n",
       "6                 Leptospira borgpetersenii (bacteria)   \n",
       "7                 Leptospira borgpetersenii (bacteria)   \n",
       "8                    Leptospira interrogans (bacteria)   \n",
       "9                    Leptospira interrogans (bacteria)   \n",
       "10                   Leptospira interrogans (bacteria)   \n",
       "11                   Leptospira interrogans (bacteria)   \n",
       "12                   Leptospira interrogans (bacteria)   \n",
       "13                   Leptospira interrogans (bacteria)   \n",
       "14                Leptospira borgpetersenii (bacteria)   \n",
       "15                   Leptospira interrogans (bacteria)   \n",
       "16                   Leptospira interrogans (bacteria)   \n",
       "17                Leptospira borgpetersenii (bacteria)   \n",
       "18                Leptospira borgpetersenii (bacteria)   \n",
       "19                   Leptospira interrogans (bacteria)   \n",
       "20                   Leptospira interrogans (bacteria)   \n",
       "21   Leptospira interrogans serovar Copenhageni str...   \n",
       "22   Leptospira borgpetersenii serovar Hardjo-bovis...   \n",
       "23   Leptospira borgpetersenii serovar Hardjo-bovis...   \n",
       "24   Leptospira biflexa serovar Patoc strain 'Patoc...   \n",
       "25   Leptospira biflexa serovar Patoc strain 'Patoc...   \n",
       "26   Leptospira interrogans serovar Lai str. 56601 ...   \n",
       "27   Leptospira interrogans serovar Copenhageni str...   \n",
       "28   Leptospira interrogans serovar Copenhageni str...   \n",
       "29   Leptospira interrogans serovar Copenhageni str...   \n",
       "..                                                 ...   \n",
       "621               Leptospira dzoumogneensis (bacteria)   \n",
       "622               Leptospira bandrabouensis (bacteria)   \n",
       "623                     Leptospira levettii (bacteria)   \n",
       "624                       Leptospira meyeri (bacteria)   \n",
       "625                     Leptospira levettii (bacteria)   \n",
       "626                       Leptospira idonii (bacteria)   \n",
       "627                Leptospira ilyithenensis (bacteria)   \n",
       "628                     Leptospira levettii (bacteria)   \n",
       "629                     Leptospira fluminis (bacteria)   \n",
       "630                  Leptospira interrogans (bacteria)   \n",
       "631                  Leptospira interrogans (bacteria)   \n",
       "632                  Leptospira interrogans (bacteria)   \n",
       "633                     Leptospira noguchii (bacteria)   \n",
       "634               Leptospira borgpetersenii (bacteria)   \n",
       "635               Leptospira borgpetersenii (bacteria)   \n",
       "636                  Leptospira interrogans (bacteria)   \n",
       "637                  Leptospira interrogans (bacteria)   \n",
       "638                  Leptospira interrogans (bacteria)   \n",
       "639                  Leptospira interrogans (bacteria)   \n",
       "640                       Leptospira weilii (bacteria)   \n",
       "641                       Leptospira weilii (bacteria)   \n",
       "642   Leptospira interrogans serovar Hardjo (bacteria)   \n",
       "643  Leptospira interrogans serovar Geyaweera (bact...   \n",
       "644  Leptospira interrogans serovar Weerasinghe (ba...   \n",
       "645  Leptospira interrogans serovar Copenhageni (ba...   \n",
       "646  Leptospira interrogans serovar Canicola (bacte...   \n",
       "647  Leptospira interrogans serovar Canicola (bacte...   \n",
       "648  Leptospira interrogans serovar Copenhageni (ba...   \n",
       "649                  Leptospira interrogans (bacteria)   \n",
       "650  Leptospira interrogans serovar Manilae (bacteria)   \n",
       "\n",
       "                       species  scaffoldN50  \\\n",
       "0       Leptospira interrogans      4218946   \n",
       "1       Leptospira interrogans      4208129   \n",
       "2               Leptospira sp.        65100   \n",
       "3    Leptospira borgpetersenii      3580756   \n",
       "4    Leptospira borgpetersenii      3579904   \n",
       "5    Leptospira borgpetersenii      3580481   \n",
       "6    Leptospira borgpetersenii      3513910   \n",
       "7    Leptospira borgpetersenii      3581767   \n",
       "8       Leptospira interrogans      4233607   \n",
       "9       Leptospira interrogans      4213733   \n",
       "10      Leptospira interrogans      4214555   \n",
       "11      Leptospira interrogans      4206042   \n",
       "12      Leptospira interrogans      4213648   \n",
       "13      Leptospira interrogans      4214175   \n",
       "14   Leptospira borgpetersenii      3601002   \n",
       "15      Leptospira interrogans      4214328   \n",
       "16      Leptospira interrogans      4212939   \n",
       "17   Leptospira borgpetersenii      3580531   \n",
       "18   Leptospira borgpetersenii      3581013   \n",
       "19      Leptospira interrogans      4213431   \n",
       "20      Leptospira interrogans      4223477   \n",
       "21      Leptospira interrogans      4277185   \n",
       "22   Leptospira borgpetersenii      3614446   \n",
       "23   Leptospira borgpetersenii      3576473   \n",
       "24          Leptospira biflexa      3603977   \n",
       "25          Leptospira biflexa      3599677   \n",
       "26      Leptospira interrogans      4338762   \n",
       "27      Leptospira interrogans        19190   \n",
       "28      Leptospira interrogans        27585   \n",
       "29      Leptospira interrogans        29809   \n",
       "..                         ...          ...   \n",
       "621  Leptospira dzoumogneensis       585855   \n",
       "622  Leptospira bandrabouensis      2180649   \n",
       "623        Leptospira levettii       884966   \n",
       "624          Leptospira meyeri      1004763   \n",
       "625        Leptospira levettii      2751354   \n",
       "626          Leptospira idonii       195150   \n",
       "627   Leptospira ilyithenensis       266913   \n",
       "628        Leptospira levettii      2854332   \n",
       "629        Leptospira fluminis       325090   \n",
       "630     Leptospira interrogans      4306144   \n",
       "631     Leptospira interrogans      4293951   \n",
       "632     Leptospira interrogans      4497709   \n",
       "633        Leptospira noguchii        15273   \n",
       "634  Leptospira borgpetersenii        35480   \n",
       "635  Leptospira borgpetersenii        34743   \n",
       "636     Leptospira interrogans        41872   \n",
       "637     Leptospira interrogans        49707   \n",
       "638     Leptospira interrogans        38918   \n",
       "639     Leptospira interrogans        41894   \n",
       "640          Leptospira weilii      3978120   \n",
       "641          Leptospira weilii      3965053   \n",
       "642     Leptospira interrogans      4355568   \n",
       "643     Leptospira interrogans      3284430   \n",
       "644     Leptospira interrogans      4266484   \n",
       "645     Leptospira interrogans        83218   \n",
       "646     Leptospira interrogans      4259066   \n",
       "647     Leptospira interrogans      4255595   \n",
       "648     Leptospira interrogans        89408   \n",
       "649     Leptospira interrogans       133191   \n",
       "650     Leptospira interrogans      4614703   \n",
       "\n",
       "                                             Institue  Assembly_status  \\\n",
       "0                     Universidade Federal de Pelotas       Chromosome   \n",
       "1                     Universidade Federal de Pelotas       Chromosome   \n",
       "2          University of Illinois at Urbana-Champaign           Contig   \n",
       "3                                       INDICASAT AIP       Chromosome   \n",
       "4                                       INDICASAT AIP       Chromosome   \n",
       "5                                       INDICASAT AIP       Chromosome   \n",
       "6                                       INDICASAT AIP       Chromosome   \n",
       "7                                       INDICASAT AIP       Chromosome   \n",
       "8                                       INDICASAT AIP       Chromosome   \n",
       "9                                       INDICASAT AIP       Chromosome   \n",
       "10                                      INDICASAT AIP       Chromosome   \n",
       "11                                      INDICASAT AIP       Chromosome   \n",
       "12                                      INDICASAT AIP       Chromosome   \n",
       "13                                      INDICASAT AIP       Chromosome   \n",
       "14                                      INDICASAT AIP       Chromosome   \n",
       "15                                      INDICASAT AIP       Chromosome   \n",
       "16                                      INDICASAT AIP       Chromosome   \n",
       "17                                      INDICASAT AIP       Chromosome   \n",
       "18                                      INDICASAT AIP       Chromosome   \n",
       "19                                      INDICASAT AIP       Chromosome   \n",
       "20                                      INDICASAT AIP       Chromosome   \n",
       "21                Sao Paulo state (Brazil) Consortium  Complete Genome   \n",
       "22                                  Monash University  Complete Genome   \n",
       "23                                  Monash University  Complete Genome   \n",
       "24                                  Monash University  Complete Genome   \n",
       "25                                   Institut Pasteur  Complete Genome   \n",
       "26                     Chinese National HGC, Shanghai  Complete Genome   \n",
       "27                          J. Craig Venter Institute         Scaffold   \n",
       "28                          J. Craig Venter Institute         Scaffold   \n",
       "29                          J. Craig Venter Institute         Scaffold   \n",
       "..                                                ...              ...   \n",
       "621                     INRS-Institut Armand-Frappier           Contig   \n",
       "622                     INRS-Institut Armand-Frappier           Contig   \n",
       "623                     INRS-Institut Armand-Frappier           Contig   \n",
       "624                     INRS-Institut Armand-Frappier           Contig   \n",
       "625                     INRS-Institut Armand-Frappier           Contig   \n",
       "626                     INRS-Institut Armand-Frappier           Contig   \n",
       "627                     INRS-Institut Armand-Frappier           Contig   \n",
       "628                     INRS-Institut Armand-Frappier           Contig   \n",
       "629                     INRS-Institut Armand-Frappier           Contig   \n",
       "630                  Rajarata University of Sri lanka  Complete Genome   \n",
       "631                  Rajarata University of Sri lanka  Complete Genome   \n",
       "632                  Rajarata University of Sri lanka  Complete Genome   \n",
       "633                   INRS - Institut Armand-Frappier           Contig   \n",
       "634                   INRS - Institut Armand-Frappier           Contig   \n",
       "635                   INRS - Institut Armand-Frappier           Contig   \n",
       "636                   INRS - Institut Armand-Frappier           Contig   \n",
       "637                   INRS - Institut Armand-Frappier           Contig   \n",
       "638                   INRS - Institut Armand-Frappier           Contig   \n",
       "639                   INRS - Institut Armand-Frappier           Contig   \n",
       "640                          Chulalongkorn University  Complete Genome   \n",
       "641                          Chulalongkorn University  Complete Genome   \n",
       "642                         Londrina State University  Complete Genome   \n",
       "643                                              JCVI           Contig   \n",
       "644                                              JCVI           Contig   \n",
       "645                           Oswaldo Cruz Foundation         Scaffold   \n",
       "646  Chinese National Human Genome Center at Shanghai  Complete Genome   \n",
       "647  Chinese National Human Genome Center at Shanghai  Complete Genome   \n",
       "648                           Oswaldo Cruz Foundation         Scaffold   \n",
       "649                      DOE - JOINT GENOME INSTITUTE         Scaffold   \n",
       "650                                  INSTITUT PASTEUR       Chromosome   \n",
       "\n",
       "      bioproject   L50  Duplication ratio  ...  Misassembled contigs length  \\\n",
       "0    PRJNA376496   1.0              1.003  ...                    4570695.0   \n",
       "1    PRJNA376501   1.0              1.007  ...                    4560160.0   \n",
       "2    PRJNA323575  15.0              1.000  ...                          0.0   \n",
       "3    PRJNA543109   1.0              1.004  ...                    3887834.0   \n",
       "4    PRJNA543109   1.0              1.004  ...                    3887110.0   \n",
       "5    PRJNA543109   1.0              1.004  ...                    3887691.0   \n",
       "6    PRJNA543109   1.0              1.004  ...                    3821116.0   \n",
       "7    PRJNA543109   1.0              1.004  ...                    3888971.0   \n",
       "8    PRJNA543109   1.0              1.007  ...                    4583691.0   \n",
       "9    PRJNA543109   1.0              1.003  ...                    4563643.0   \n",
       "10   PRJNA543109   1.0              1.003  ...                    4564465.0   \n",
       "11   PRJNA543109   1.0              1.003  ...                    4556294.0   \n",
       "12   PRJNA543109   1.0              1.003  ...                    4563558.0   \n",
       "13   PRJNA543109   1.0              1.003  ...                    4564085.0   \n",
       "14   PRJNA543109   1.0              1.005  ...                    3908796.0   \n",
       "15   PRJNA543109   1.0              1.002  ...                    4564238.0   \n",
       "16   PRJNA543109   1.0              1.003  ...                    4563191.0   \n",
       "17   PRJNA543109   1.0              1.004  ...                    3887609.0   \n",
       "18   PRJNA543109   1.0              1.004  ...                    3888127.0   \n",
       "19   PRJNA543109   1.0              1.003  ...                    4563683.0   \n",
       "20   PRJNA543109   1.0              1.003  ...                    4573647.0   \n",
       "21    PRJNA10687   1.0              1.004  ...                    4627366.0   \n",
       "22    PRJNA16146   1.0              2.073  ...                          0.0   \n",
       "23    PRJNA16148   1.0              2.052  ...                          0.0   \n",
       "24    PRJNA16153   1.0              2.000  ...                          0.0   \n",
       "25    PRJNA20133   1.0              2.000  ...                          0.0   \n",
       "26      PRJNA293   1.0              1.000  ...                          0.0   \n",
       "27    PRJNA65081  74.0              1.005  ...                     943347.0   \n",
       "28    PRJNA65079  49.0              1.004  ...                    1147421.0   \n",
       "29    PRJNA65077  50.0              1.004  ...                    1482896.0   \n",
       "..           ...   ...                ...  ...                          ...   \n",
       "621  PRJNA480513   3.0              1.135  ...                          0.0   \n",
       "622  PRJNA480513   1.0              1.000  ...                          0.0   \n",
       "623  PRJNA480513   2.0              1.635  ...                          0.0   \n",
       "624  PRJNA480513   2.0              1.000  ...                          0.0   \n",
       "625  PRJNA480513   1.0              1.776  ...                          0.0   \n",
       "626  PRJNA480513   7.0              1.000  ...                          0.0   \n",
       "627  PRJNA480513   5.0              1.000  ...                          0.0   \n",
       "628  PRJNA480513   1.0              1.239  ...                          0.0   \n",
       "629  PRJNA480513   4.0              1.146  ...                          0.0   \n",
       "630  PRJNA528695   1.0              1.010  ...                    4661889.0   \n",
       "631  PRJNA528695   1.0              1.004  ...                    4651137.0   \n",
       "632  PRJNA528695   1.0              1.012  ...                    4858092.0   \n",
       "633  PRJNA543681  95.0              1.235  ...                          0.0   \n",
       "634  PRJNA543681  30.0              1.339  ...                          0.0   \n",
       "635  PRJNA543681  28.0              1.316  ...                          0.0   \n",
       "636  PRJNA543681  31.0              1.007  ...                    2217479.0   \n",
       "637  PRJNA543681  29.0              1.007  ...                    2050612.0   \n",
       "638  PRJNA543681  38.0              1.008  ...                    2055025.0   \n",
       "639  PRJNA543681  31.0              1.007  ...                    2265076.0   \n",
       "640  PRJNA545909   1.0              1.370  ...                          0.0   \n",
       "641  PRJNA545906   1.0              1.370  ...                          0.0   \n",
       "642  PRJNA560522   1.0              1.022  ...                    4705503.0   \n",
       "643  PRJNA401006   1.0              1.017  ...                    4695993.0   \n",
       "644  PRJNA401006   1.0              1.053  ...                    4947504.0   \n",
       "645  PRJNA560059  19.0              1.003  ...                    2853158.0   \n",
       "646  PRJNA574874   1.0              1.008  ...                    4616103.0   \n",
       "647  PRJNA574874   1.0              1.008  ...                    4612880.0   \n",
       "648  PRJNA560059  18.0              1.003  ...                    3040784.0   \n",
       "649   PRJEB18875  10.0              1.005  ...                    3949662.0   \n",
       "650   PRJEB23342   1.0              1.003  ...                    4614703.0   \n",
       "\n",
       "     Reference length  # misassembled contigs      NGA75       NA75  LGA75  \\\n",
       "0           4698134.0                     2.0    30352.0    32897.0   52.0   \n",
       "1           4698134.0                     2.0    28968.0    30585.0   54.0   \n",
       "2           4698134.0                     0.0        NaN        NaN    NaN   \n",
       "3           3931782.0                     2.0    15022.0    15795.0   80.0   \n",
       "4           3931782.0                     2.0    14988.0    15170.0   81.0   \n",
       "5           3931782.0                     2.0    15022.0    15795.0   80.0   \n",
       "6           3931782.0                     2.0    14571.0    15170.0   83.0   \n",
       "7           3931782.0                     2.0    14988.0    15170.0   81.0   \n",
       "8           4698134.0                     2.0    60938.0    61586.0   26.0   \n",
       "9           4698134.0                     2.0    60640.0    61945.0   30.0   \n",
       "10          4698134.0                     2.0    61945.0    64756.0   28.0   \n",
       "11          4698134.0                     2.0    61630.0    64047.0   28.0   \n",
       "12          4698134.0                     2.0    61630.0    64047.0   30.0   \n",
       "13          4698134.0                     2.0    61630.0    63947.0   30.0   \n",
       "14          3931782.0                     2.0    15022.0    15169.0   82.0   \n",
       "15          4698134.0                     2.0    61630.0    63947.0   30.0   \n",
       "16          4698134.0                     2.0    61630.0    64047.0   29.0   \n",
       "17          3931782.0                     2.0    14988.0    15795.0   81.0   \n",
       "18          3931782.0                     2.0    15022.0    15883.0   79.0   \n",
       "19          4698134.0                     2.0    61630.0    63947.0   28.0   \n",
       "20          4698134.0                     2.0    63194.0    66836.0   26.0   \n",
       "21          4698134.0                     2.0    46921.0    47908.0   38.0   \n",
       "22          4698134.0                     0.0        NaN        NaN    NaN   \n",
       "23          4698134.0                     0.0        NaN        NaN    NaN   \n",
       "24          4698134.0                     0.0        NaN        NaN    NaN   \n",
       "25          4698134.0                     0.0        NaN        NaN    NaN   \n",
       "26          4698134.0                     0.0  4338762.0  4338762.0    1.0   \n",
       "27          4698134.0                    50.0     7346.0     8395.0  198.0   \n",
       "28          4698134.0                    41.0    12247.0    13704.0  128.0   \n",
       "29          4698134.0                    54.0    11357.0    12984.0  132.0   \n",
       "..                ...                     ...        ...        ...    ...   \n",
       "621         4698134.0                     0.0        NaN        NaN    NaN   \n",
       "622         4698134.0                     0.0        NaN        NaN    NaN   \n",
       "623         4698134.0                     0.0        NaN        NaN    NaN   \n",
       "624         4698134.0                     0.0        NaN        NaN    NaN   \n",
       "625         4698134.0                     0.0        NaN        NaN    NaN   \n",
       "626         4698134.0                     0.0        NaN        NaN    NaN   \n",
       "627         4698134.0                     0.0        NaN        NaN    NaN   \n",
       "628         4698134.0                     0.0        NaN        NaN    NaN   \n",
       "629         4698134.0                     0.0        NaN        NaN    NaN   \n",
       "630         4698134.0                     2.0    28022.0    26733.0   56.0   \n",
       "631         4698134.0                     2.0    33799.0    34493.0   54.0   \n",
       "632         4698134.0                     2.0    31632.0    20236.0   50.0   \n",
       "633         4698134.0                     0.0        NaN        NaN    NaN   \n",
       "634         4698134.0                     0.0        NaN        NaN    NaN   \n",
       "635         4698134.0                     0.0        NaN        NaN    NaN   \n",
       "636         4698134.0                    59.0    12932.0    14831.0  109.0   \n",
       "637         4698134.0                    52.0    13590.0    14976.0  106.0   \n",
       "638         4698134.0                    64.0    12336.0    14091.0  119.0   \n",
       "639         4698134.0                    58.0    12932.0    14191.0  109.0   \n",
       "640         4698134.0                     0.0        NaN        NaN    NaN   \n",
       "641         4698134.0                     0.0        NaN        NaN    NaN   \n",
       "642         4698134.0                     2.0    35465.0    35198.0   47.0   \n",
       "643         4698134.0                     6.0    29455.0    16143.0   59.0   \n",
       "644         4698134.0                     5.0    29375.0    22847.0   62.0   \n",
       "645         4698134.0                    41.0    23488.0    27692.0   66.0   \n",
       "646         4698134.0                     2.0    27538.0    26688.0   58.0   \n",
       "647         4698134.0                     2.0    27538.0    26688.0   59.0   \n",
       "648         4698134.0                    38.0    25000.0    26527.0   66.0   \n",
       "649         4698134.0                    37.0    35959.0    37810.0   50.0   \n",
       "650         4698134.0                     1.0    22946.0    23335.0   77.0   \n",
       "\n",
       "          NA50  LGA50   LA75  LA50  \n",
       "0      62085.0   24.0   49.0  23.0  \n",
       "1      60422.0   24.0   51.0  23.0  \n",
       "2          NaN    NaN    NaN   NaN  \n",
       "3      34042.0   37.0   78.0  37.0  \n",
       "4      34043.0   37.0   79.0  37.0  \n",
       "5      34043.0   37.0   78.0  37.0  \n",
       "6      34725.0   37.0   77.0  36.0  \n",
       "7      34043.0   37.0   79.0  37.0  \n",
       "8     132556.0   12.0   25.0  11.0  \n",
       "9      99422.0   14.0   28.0  13.0  \n",
       "10    122749.0   13.0   26.0  12.0  \n",
       "11    114234.0   13.0   27.0  13.0  \n",
       "12    110881.0   14.0   28.0  14.0  \n",
       "13    110881.0   14.0   28.0  14.0  \n",
       "14     34043.0   38.0   81.0  37.0  \n",
       "15    110881.0   14.0   28.0  14.0  \n",
       "16    110881.0   14.0   28.0  14.0  \n",
       "17     34043.0   37.0   78.0  37.0  \n",
       "18     34725.0   37.0   77.0  36.0  \n",
       "19    114234.0   13.0   27.0  13.0  \n",
       "20    115598.0   13.0   25.0  12.0  \n",
       "21     80954.0   19.0   37.0  19.0  \n",
       "22         NaN    NaN    NaN   NaN  \n",
       "23         NaN    NaN    NaN   NaN  \n",
       "24         NaN    NaN    NaN   NaN  \n",
       "25         NaN    NaN    NaN   NaN  \n",
       "26   4338762.0    1.0    1.0   1.0  \n",
       "27     16502.0   91.0  182.0  86.0  \n",
       "28     23020.0   60.0  119.0  56.0  \n",
       "29     26118.0   61.0  121.0  57.0  \n",
       "..         ...    ...    ...   ...  \n",
       "621        NaN    NaN    NaN   NaN  \n",
       "622        NaN    NaN    NaN   NaN  \n",
       "623        NaN    NaN    NaN   NaN  \n",
       "624        NaN    NaN    NaN   NaN  \n",
       "625        NaN    NaN    NaN   NaN  \n",
       "626        NaN    NaN    NaN   NaN  \n",
       "627        NaN    NaN    NaN   NaN  \n",
       "628        NaN    NaN    NaN   NaN  \n",
       "629        NaN    NaN    NaN   NaN  \n",
       "630    52332.0   26.0   58.0  27.0  \n",
       "631    54269.0   27.0   52.0  27.0  \n",
       "632    53490.0   24.0   62.0  27.0  \n",
       "633        NaN    NaN    NaN   NaN  \n",
       "634        NaN    NaN    NaN   NaN  \n",
       "635        NaN    NaN    NaN   NaN  \n",
       "636    29836.0   47.0  100.0  44.0  \n",
       "637    31296.0   45.0   97.0  42.0  \n",
       "638    27872.0   53.0  109.0  50.0  \n",
       "639    29203.0   46.0  100.0  43.0  \n",
       "640        NaN    NaN    NaN   NaN  \n",
       "641        NaN    NaN    NaN   NaN  \n",
       "642    66852.0   22.0   48.0  23.0  \n",
       "643    46841.0   29.0   80.0  35.0  \n",
       "644    45717.0   31.0   72.0  35.0  \n",
       "645    48492.0   31.0   62.0  29.0  \n",
       "646    50605.0   29.0   60.0  30.0  \n",
       "647    50648.0   30.0   61.0  30.0  \n",
       "648    48083.0   32.0   63.0  30.0  \n",
       "649    57457.0   24.0   48.0  23.0  \n",
       "650    38033.0   37.0   75.0  36.0  \n",
       "\n",
       "[651 rows x 61 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "biosamples = pd.merge(full_cov,biosamples)\n",
    "# take only the isolates from the query species\n",
    "# biosamples = biosamples[biosamples[\"biosample_acc\"].index.isin(species_acc.index)]\n",
    "\n",
    "biosamples\n",
    "# biosamples[\"coverage\"]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>biosample_acc</th>\n",
       "      <th>coverage</th>\n",
       "      <th># contigs</th>\n",
       "      <th>Total length</th>\n",
       "      <th>N50</th>\n",
       "      <th># misassemblies</th>\n",
       "      <th># N's per 100 kbp</th>\n",
       "      <th>species</th>\n",
       "      <th>organism</th>\n",
       "      <th>Institue</th>\n",
       "      <th>Assembly_status</th>\n",
       "      <th>bioproject</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SAMN06434252</td>\n",
       "      <td>30.0000</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4570695.0</td>\n",
       "      <td>4218946.0</td>\n",
       "      <td>128.0</td>\n",
       "      <td>207.89</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>Leptospira interrogans serovar Canicola (bacte...</td>\n",
       "      <td>Universidade Federal de Pelotas</td>\n",
       "      <td>Chromosome</td>\n",
       "      <td>PRJNA376496</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SAMN06434273</td>\n",
       "      <td>30.0000</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4560160.0</td>\n",
       "      <td>4208129.0</td>\n",
       "      <td>130.0</td>\n",
       "      <td>612.08</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>Leptospira interrogans serovar Canicola (bacte...</td>\n",
       "      <td>Universidade Federal de Pelotas</td>\n",
       "      <td>Chromosome</td>\n",
       "      <td>PRJNA376501</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>SAMN02603847</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4627366.0</td>\n",
       "      <td>4277185.0</td>\n",
       "      <td>98.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>Leptospira interrogans serovar Copenhageni str...</td>\n",
       "      <td>Sao Paulo state (Brazil) Consortium</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>PRJNA10687</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>SAMN02603615</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>2.0</td>\n",
       "      <td>3931782.0</td>\n",
       "      <td>3614446.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Leptospira borgpetersenii</td>\n",
       "      <td>Leptospira borgpetersenii serovar Hardjo-bovis...</td>\n",
       "      <td>Monash University</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>PRJNA16146</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>SAMN02603616</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>2.0</td>\n",
       "      <td>3876235.0</td>\n",
       "      <td>3576473.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Leptospira borgpetersenii</td>\n",
       "      <td>Leptospira borgpetersenii serovar Hardjo-bovis...</td>\n",
       "      <td>Monash University</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>PRJNA16148</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>SAMN02603617</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3956089.0</td>\n",
       "      <td>3603977.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Leptospira biflexa</td>\n",
       "      <td>Leptospira biflexa serovar Patoc strain 'Patoc...</td>\n",
       "      <td>Monash University</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>PRJNA16153</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>SAMN02603347</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3951448.0</td>\n",
       "      <td>3599677.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Leptospira biflexa</td>\n",
       "      <td>Leptospira biflexa serovar Patoc strain 'Patoc...</td>\n",
       "      <td>Institut Pasteur</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>PRJNA20133</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>SAMN02603127</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4698134.0</td>\n",
       "      <td>4338762.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>Leptospira interrogans serovar Lai str. 56601 ...</td>\n",
       "      <td>Chinese National HGC, Shanghai</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>PRJNA293</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>SAMN00255265</td>\n",
       "      <td>34.4000</td>\n",
       "      <td>16.0</td>\n",
       "      <td>4705262.0</td>\n",
       "      <td>617754.0</td>\n",
       "      <td>96.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>Leptospira interrogans serovar Canicola str. L...</td>\n",
       "      <td>J. Craig Venter Institute</td>\n",
       "      <td>Contig</td>\n",
       "      <td>PRJNA65063</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>SAMN02436336</td>\n",
       "      <td>51.0000</td>\n",
       "      <td>229.0</td>\n",
       "      <td>4679663.0</td>\n",
       "      <td>37187.0</td>\n",
       "      <td>85.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>Leptospira interrogans serovar Djasiman str. L...</td>\n",
       "      <td>J. Craig Venter Institute</td>\n",
       "      <td>Contig</td>\n",
       "      <td>PRJNA65061</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>SAMN02436629</td>\n",
       "      <td>53.6000</td>\n",
       "      <td>43.0</td>\n",
       "      <td>4741083.0</td>\n",
       "      <td>250073.0</td>\n",
       "      <td>136.0</td>\n",
       "      <td>0.04</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>Leptospira interrogans str. C10069 (bacteria)</td>\n",
       "      <td>J. Craig Venter Institute</td>\n",
       "      <td>Contig</td>\n",
       "      <td>PRJNA65055</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>SAMN02436466</td>\n",
       "      <td>53.6000</td>\n",
       "      <td>84.0</td>\n",
       "      <td>4813972.0</td>\n",
       "      <td>127002.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.02</td>\n",
       "      <td>Leptospira noguchii</td>\n",
       "      <td>Leptospira noguchii str. 2006001870 (bacteria)</td>\n",
       "      <td>J. Craig Venter Institute</td>\n",
       "      <td>Contig</td>\n",
       "      <td>PRJNA65053</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>SAMN02436545</td>\n",
       "      <td>52.8000</td>\n",
       "      <td>35.0</td>\n",
       "      <td>3914343.0</td>\n",
       "      <td>399563.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>Leptospira santarosai</td>\n",
       "      <td>Leptospira santarosai str. 2000027870 (bacteria)</td>\n",
       "      <td>J. Craig Venter Institute</td>\n",
       "      <td>Contig</td>\n",
       "      <td>PRJNA65051</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>SAMN02436335</td>\n",
       "      <td>63.7000</td>\n",
       "      <td>84.0</td>\n",
       "      <td>4370034.0</td>\n",
       "      <td>148695.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Leptospira weilii</td>\n",
       "      <td>Leptospira weilii str. 2006001853 (bacteria)</td>\n",
       "      <td>J. Craig Venter Institute</td>\n",
       "      <td>Contig</td>\n",
       "      <td>PRJNA65047</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>SAMN02436628</td>\n",
       "      <td>57.0000</td>\n",
       "      <td>71.0</td>\n",
       "      <td>4918776.0</td>\n",
       "      <td>130328.0</td>\n",
       "      <td>150.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>Leptospira interrogans str. 2002000621 (bacteria)</td>\n",
       "      <td>J. Craig Venter Institute</td>\n",
       "      <td>Contig</td>\n",
       "      <td>PRJNA65045</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>SAMN02436547</td>\n",
       "      <td>62.2000</td>\n",
       "      <td>42.0</td>\n",
       "      <td>4580695.0</td>\n",
       "      <td>198881.0</td>\n",
       "      <td>153.0</td>\n",
       "      <td>0.07</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>Leptospira interrogans serovar Pomona str. Pom...</td>\n",
       "      <td>J. Craig Venter Institute</td>\n",
       "      <td>Contig</td>\n",
       "      <td>PRJNA65043</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>SAMN02436546</td>\n",
       "      <td>62.0000</td>\n",
       "      <td>40.0</td>\n",
       "      <td>4660976.0</td>\n",
       "      <td>198307.0</td>\n",
       "      <td>122.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>Leptospira interrogans serovar Bulgarica str. ...</td>\n",
       "      <td>J. Craig Venter Institute</td>\n",
       "      <td>Contig</td>\n",
       "      <td>PRJNA65041</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>SAMN00254365</td>\n",
       "      <td>30.1100</td>\n",
       "      <td>252.0</td>\n",
       "      <td>4545360.0</td>\n",
       "      <td>30507.0</td>\n",
       "      <td>54.0</td>\n",
       "      <td>16.17</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>Leptospira interrogans serovar Copenhageni str...</td>\n",
       "      <td>J. Craig Venter Institute</td>\n",
       "      <td>Scaffold</td>\n",
       "      <td>PRJNA63819</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>SAMN00254363</td>\n",
       "      <td>39.6900</td>\n",
       "      <td>242.0</td>\n",
       "      <td>4545399.0</td>\n",
       "      <td>33096.0</td>\n",
       "      <td>48.0</td>\n",
       "      <td>19.76</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>Leptospira interrogans serovar Copenhageni str...</td>\n",
       "      <td>J. Craig Venter Institute</td>\n",
       "      <td>Scaffold</td>\n",
       "      <td>PRJNA63817</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>SAMN00254358</td>\n",
       "      <td>46.6800</td>\n",
       "      <td>232.0</td>\n",
       "      <td>4548800.0</td>\n",
       "      <td>36256.0</td>\n",
       "      <td>49.0</td>\n",
       "      <td>13.65</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>Leptospira interrogans serovar Copenhageni str...</td>\n",
       "      <td>J. Craig Venter Institute</td>\n",
       "      <td>Scaffold</td>\n",
       "      <td>PRJNA63805</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>SAMN00254352</td>\n",
       "      <td>47.3200</td>\n",
       "      <td>280.0</td>\n",
       "      <td>4544687.0</td>\n",
       "      <td>27958.0</td>\n",
       "      <td>39.0</td>\n",
       "      <td>25.00</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>Leptospira interrogans serovar Copenhageni str...</td>\n",
       "      <td>J. Craig Venter Institute</td>\n",
       "      <td>Scaffold</td>\n",
       "      <td>PRJNA63793</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>SAMN00254342</td>\n",
       "      <td>49.1600</td>\n",
       "      <td>236.0</td>\n",
       "      <td>4547942.0</td>\n",
       "      <td>37545.0</td>\n",
       "      <td>47.0</td>\n",
       "      <td>16.40</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>Leptospira interrogans serovar Copenhageni str...</td>\n",
       "      <td>J. Craig Venter Institute</td>\n",
       "      <td>Scaffold</td>\n",
       "      <td>PRJNA63771</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>SAMN00254336</td>\n",
       "      <td>65.2700</td>\n",
       "      <td>236.0</td>\n",
       "      <td>4550206.0</td>\n",
       "      <td>35609.0</td>\n",
       "      <td>46.0</td>\n",
       "      <td>17.74</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>Leptospira interrogans serovar Copenhageni str...</td>\n",
       "      <td>J. Craig Venter Institute</td>\n",
       "      <td>Scaffold</td>\n",
       "      <td>PRJNA63759</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>SAMN00254335</td>\n",
       "      <td>74.6100</td>\n",
       "      <td>235.0</td>\n",
       "      <td>4547515.0</td>\n",
       "      <td>32986.0</td>\n",
       "      <td>53.0</td>\n",
       "      <td>14.69</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>Leptospira interrogans serovar Copenhageni str...</td>\n",
       "      <td>J. Craig Venter Institute</td>\n",
       "      <td>Scaffold</td>\n",
       "      <td>PRJNA63757</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81</th>\n",
       "      <td>SAMN00254334</td>\n",
       "      <td>72.4500</td>\n",
       "      <td>233.0</td>\n",
       "      <td>4549651.0</td>\n",
       "      <td>37915.0</td>\n",
       "      <td>44.0</td>\n",
       "      <td>17.41</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>Leptospira interrogans serovar Copenhageni str...</td>\n",
       "      <td>J. Craig Venter Institute</td>\n",
       "      <td>Scaffold</td>\n",
       "      <td>PRJNA63755</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>SAMN00254333</td>\n",
       "      <td>86.4700</td>\n",
       "      <td>226.0</td>\n",
       "      <td>4546512.0</td>\n",
       "      <td>36023.0</td>\n",
       "      <td>45.0</td>\n",
       "      <td>15.15</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>Leptospira interrogans serovar Copenhageni str...</td>\n",
       "      <td>J. Craig Venter Institute</td>\n",
       "      <td>Scaffold</td>\n",
       "      <td>PRJNA63753</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>SAMN00254332</td>\n",
       "      <td>83.0500</td>\n",
       "      <td>222.0</td>\n",
       "      <td>4545322.0</td>\n",
       "      <td>35769.0</td>\n",
       "      <td>53.0</td>\n",
       "      <td>13.16</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>Leptospira interrogans serovar Copenhageni str...</td>\n",
       "      <td>J. Craig Venter Institute</td>\n",
       "      <td>Scaffold</td>\n",
       "      <td>PRJNA63751</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>SAMN00254330</td>\n",
       "      <td>104.1200</td>\n",
       "      <td>234.0</td>\n",
       "      <td>4545200.0</td>\n",
       "      <td>38517.0</td>\n",
       "      <td>44.0</td>\n",
       "      <td>15.05</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>Leptospira interrogans serovar Copenhageni str...</td>\n",
       "      <td>J. Craig Venter Institute</td>\n",
       "      <td>Scaffold</td>\n",
       "      <td>PRJNA63747</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86</th>\n",
       "      <td>SAMN00254329</td>\n",
       "      <td>98.7000</td>\n",
       "      <td>224.0</td>\n",
       "      <td>4548549.0</td>\n",
       "      <td>34635.0</td>\n",
       "      <td>46.0</td>\n",
       "      <td>19.39</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>Leptospira interrogans serovar Copenhageni str...</td>\n",
       "      <td>J. Craig Venter Institute</td>\n",
       "      <td>Scaffold</td>\n",
       "      <td>PRJNA63745</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>SAMN00254328</td>\n",
       "      <td>92.6100</td>\n",
       "      <td>244.0</td>\n",
       "      <td>4549345.0</td>\n",
       "      <td>32537.0</td>\n",
       "      <td>47.0</td>\n",
       "      <td>16.05</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>Leptospira interrogans serovar Copenhageni str...</td>\n",
       "      <td>J. Craig Venter Institute</td>\n",
       "      <td>Scaffold</td>\n",
       "      <td>PRJNA63743</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>620</th>\n",
       "      <td>SAMN10316370</td>\n",
       "      <td>50.0000</td>\n",
       "      <td>205.0</td>\n",
       "      <td>4014286.0</td>\n",
       "      <td>60816.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Leptospira mayottensis</td>\n",
       "      <td>Leptospira mayottensis (bacteria)</td>\n",
       "      <td>INRS-Institut Armand-Frappier</td>\n",
       "      <td>Contig</td>\n",
       "      <td>PRJNA480513</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>621</th>\n",
       "      <td>SAMN10316368</td>\n",
       "      <td>50.0000</td>\n",
       "      <td>28.0</td>\n",
       "      <td>4118320.0</td>\n",
       "      <td>585855.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Leptospira dzoumogneensis</td>\n",
       "      <td>Leptospira dzoumogneensis (bacteria)</td>\n",
       "      <td>INRS-Institut Armand-Frappier</td>\n",
       "      <td>Contig</td>\n",
       "      <td>PRJNA480513</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>622</th>\n",
       "      <td>SAMN10316367</td>\n",
       "      <td>50.0000</td>\n",
       "      <td>18.0</td>\n",
       "      <td>4027080.0</td>\n",
       "      <td>2180649.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Leptospira bandrabouensis</td>\n",
       "      <td>Leptospira bandrabouensis (bacteria)</td>\n",
       "      <td>INRS-Institut Armand-Frappier</td>\n",
       "      <td>Contig</td>\n",
       "      <td>PRJNA480513</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>623</th>\n",
       "      <td>SAMN10316389</td>\n",
       "      <td>50.0000</td>\n",
       "      <td>13.0</td>\n",
       "      <td>3849801.0</td>\n",
       "      <td>884966.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Leptospira levettii</td>\n",
       "      <td>Leptospira levettii (bacteria)</td>\n",
       "      <td>INRS-Institut Armand-Frappier</td>\n",
       "      <td>Contig</td>\n",
       "      <td>PRJNA480513</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>624</th>\n",
       "      <td>SAMN10316379</td>\n",
       "      <td>50.0000</td>\n",
       "      <td>56.0</td>\n",
       "      <td>4156836.0</td>\n",
       "      <td>1004763.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Leptospira meyeri</td>\n",
       "      <td>Leptospira meyeri (bacteria)</td>\n",
       "      <td>INRS-Institut Armand-Frappier</td>\n",
       "      <td>Contig</td>\n",
       "      <td>PRJNA480513</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>625</th>\n",
       "      <td>SAMN10316374</td>\n",
       "      <td>50.0000</td>\n",
       "      <td>8.0</td>\n",
       "      <td>3889488.0</td>\n",
       "      <td>2751354.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Leptospira levettii</td>\n",
       "      <td>Leptospira levettii (bacteria)</td>\n",
       "      <td>INRS-Institut Armand-Frappier</td>\n",
       "      <td>Contig</td>\n",
       "      <td>PRJNA480513</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>626</th>\n",
       "      <td>SAMN10316364</td>\n",
       "      <td>50.0000</td>\n",
       "      <td>78.0</td>\n",
       "      <td>4086283.0</td>\n",
       "      <td>195150.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Leptospira idonii</td>\n",
       "      <td>Leptospira idonii (bacteria)</td>\n",
       "      <td>INRS-Institut Armand-Frappier</td>\n",
       "      <td>Contig</td>\n",
       "      <td>PRJNA480513</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>627</th>\n",
       "      <td>SAMN10316365</td>\n",
       "      <td>50.0000</td>\n",
       "      <td>63.0</td>\n",
       "      <td>4207523.0</td>\n",
       "      <td>266913.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Leptospira ilyithenensis</td>\n",
       "      <td>Leptospira ilyithenensis (bacteria)</td>\n",
       "      <td>INRS-Institut Armand-Frappier</td>\n",
       "      <td>Contig</td>\n",
       "      <td>PRJNA480513</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>628</th>\n",
       "      <td>SAMN10316373</td>\n",
       "      <td>50.0000</td>\n",
       "      <td>7.0</td>\n",
       "      <td>3935107.0</td>\n",
       "      <td>2854332.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Leptospira levettii</td>\n",
       "      <td>Leptospira levettii (bacteria)</td>\n",
       "      <td>INRS-Institut Armand-Frappier</td>\n",
       "      <td>Contig</td>\n",
       "      <td>PRJNA480513</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>629</th>\n",
       "      <td>SAMN10316443</td>\n",
       "      <td>50.0000</td>\n",
       "      <td>18.0</td>\n",
       "      <td>3746420.0</td>\n",
       "      <td>325090.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Leptospira fluminis</td>\n",
       "      <td>Leptospira fluminis (bacteria)</td>\n",
       "      <td>INRS-Institut Armand-Frappier</td>\n",
       "      <td>Contig</td>\n",
       "      <td>PRJNA480513</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>630</th>\n",
       "      <td>SAMN11289419</td>\n",
       "      <td>30.0000</td>\n",
       "      <td>3.0</td>\n",
       "      <td>4740090.0</td>\n",
       "      <td>4306144.0</td>\n",
       "      <td>162.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>Leptospira interrogans (bacteria)</td>\n",
       "      <td>Rajarata University of Sri lanka</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>PRJNA528695</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>631</th>\n",
       "      <td>SAMN11289420</td>\n",
       "      <td>30.0000</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4651137.0</td>\n",
       "      <td>4293951.0</td>\n",
       "      <td>131.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>Leptospira interrogans (bacteria)</td>\n",
       "      <td>Rajarata University of Sri lanka</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>PRJNA528695</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>632</th>\n",
       "      <td>SAMN11289421</td>\n",
       "      <td>30.0000</td>\n",
       "      <td>4.0</td>\n",
       "      <td>5071735.0</td>\n",
       "      <td>4497709.0</td>\n",
       "      <td>155.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>Leptospira interrogans (bacteria)</td>\n",
       "      <td>Rajarata University of Sri lanka</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>PRJNA528695</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>634</th>\n",
       "      <td>SAMN11774879</td>\n",
       "      <td>95.0000</td>\n",
       "      <td>196.0</td>\n",
       "      <td>3706445.0</td>\n",
       "      <td>35480.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Leptospira borgpetersenii</td>\n",
       "      <td>Leptospira borgpetersenii (bacteria)</td>\n",
       "      <td>INRS - Institut Armand-Frappier</td>\n",
       "      <td>Contig</td>\n",
       "      <td>PRJNA543681</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>635</th>\n",
       "      <td>SAMN11774878</td>\n",
       "      <td>96.0000</td>\n",
       "      <td>194.0</td>\n",
       "      <td>3728833.0</td>\n",
       "      <td>35794.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Leptospira borgpetersenii</td>\n",
       "      <td>Leptospira borgpetersenii (bacteria)</td>\n",
       "      <td>INRS - Institut Armand-Frappier</td>\n",
       "      <td>Contig</td>\n",
       "      <td>PRJNA543681</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>636</th>\n",
       "      <td>SAMN11774876</td>\n",
       "      <td>98.0000</td>\n",
       "      <td>213.0</td>\n",
       "      <td>4530054.0</td>\n",
       "      <td>41872.0</td>\n",
       "      <td>84.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>Leptospira interrogans (bacteria)</td>\n",
       "      <td>INRS - Institut Armand-Frappier</td>\n",
       "      <td>Contig</td>\n",
       "      <td>PRJNA543681</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>637</th>\n",
       "      <td>SAMN11774877</td>\n",
       "      <td>98.0000</td>\n",
       "      <td>202.0</td>\n",
       "      <td>4525529.0</td>\n",
       "      <td>49707.0</td>\n",
       "      <td>77.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>Leptospira interrogans (bacteria)</td>\n",
       "      <td>INRS - Institut Armand-Frappier</td>\n",
       "      <td>Contig</td>\n",
       "      <td>PRJNA543681</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>638</th>\n",
       "      <td>SAMN11774874</td>\n",
       "      <td>98.0000</td>\n",
       "      <td>226.0</td>\n",
       "      <td>4522106.0</td>\n",
       "      <td>39552.0</td>\n",
       "      <td>84.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>Leptospira interrogans (bacteria)</td>\n",
       "      <td>INRS - Institut Armand-Frappier</td>\n",
       "      <td>Contig</td>\n",
       "      <td>PRJNA543681</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>639</th>\n",
       "      <td>SAMN11774875</td>\n",
       "      <td>98.0000</td>\n",
       "      <td>211.0</td>\n",
       "      <td>4529013.0</td>\n",
       "      <td>41894.0</td>\n",
       "      <td>83.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>Leptospira interrogans (bacteria)</td>\n",
       "      <td>INRS - Institut Armand-Frappier</td>\n",
       "      <td>Contig</td>\n",
       "      <td>PRJNA543681</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>640</th>\n",
       "      <td>SAMN11936725</td>\n",
       "      <td>38.0000</td>\n",
       "      <td>3.0</td>\n",
       "      <td>4361439.0</td>\n",
       "      <td>3978120.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Leptospira weilii</td>\n",
       "      <td>Leptospira weilii (bacteria)</td>\n",
       "      <td>Chulalongkorn University</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>PRJNA545909</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>641</th>\n",
       "      <td>SAMN11936708</td>\n",
       "      <td>30.0000</td>\n",
       "      <td>3.0</td>\n",
       "      <td>4381393.0</td>\n",
       "      <td>3965053.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Leptospira weilii</td>\n",
       "      <td>Leptospira weilii (bacteria)</td>\n",
       "      <td>Chulalongkorn University</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>PRJNA545906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>642</th>\n",
       "      <td>SAMN12588099</td>\n",
       "      <td>215.0000</td>\n",
       "      <td>3.0</td>\n",
       "      <td>4742398.0</td>\n",
       "      <td>4355568.0</td>\n",
       "      <td>151.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>Leptospira interrogans serovar Hardjo (bacteria)</td>\n",
       "      <td>Londrina State University</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>PRJNA560522</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>643</th>\n",
       "      <td>SAMN08513768</td>\n",
       "      <td>31.5296</td>\n",
       "      <td>16.0</td>\n",
       "      <td>5336364.0</td>\n",
       "      <td>3284430.0</td>\n",
       "      <td>149.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>Leptospira interrogans serovar Geyaweera (bact...</td>\n",
       "      <td>JCVI</td>\n",
       "      <td>Contig</td>\n",
       "      <td>PRJNA401006</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>644</th>\n",
       "      <td>SAMN08596279</td>\n",
       "      <td>498.6260</td>\n",
       "      <td>7.0</td>\n",
       "      <td>5040606.0</td>\n",
       "      <td>4266484.0</td>\n",
       "      <td>174.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>Leptospira interrogans serovar Weerasinghe (ba...</td>\n",
       "      <td>JCVI</td>\n",
       "      <td>Contig</td>\n",
       "      <td>PRJNA401006</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>645</th>\n",
       "      <td>SAMN12567456</td>\n",
       "      <td>888.0000</td>\n",
       "      <td>112.0</td>\n",
       "      <td>4565779.0</td>\n",
       "      <td>83218.0</td>\n",
       "      <td>64.0</td>\n",
       "      <td>23.70</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>Leptospira interrogans serovar Copenhageni (ba...</td>\n",
       "      <td>Oswaldo Cruz Foundation</td>\n",
       "      <td>Scaffold</td>\n",
       "      <td>PRJNA560059</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>646</th>\n",
       "      <td>SAMN12871420</td>\n",
       "      <td>100.0000</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4758488.0</td>\n",
       "      <td>4259066.0</td>\n",
       "      <td>165.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>Leptospira interrogans serovar Canicola (bacte...</td>\n",
       "      <td>Chinese National Human Genome Center at Shanghai</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>PRJNA574874</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>647</th>\n",
       "      <td>SAMN12871448</td>\n",
       "      <td>100.0000</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4755342.0</td>\n",
       "      <td>4255595.0</td>\n",
       "      <td>166.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>Leptospira interrogans serovar Canicola (bacte...</td>\n",
       "      <td>Chinese National Human Genome Center at Shanghai</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>PRJNA574874</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>648</th>\n",
       "      <td>SAMN12567526</td>\n",
       "      <td>152.0000</td>\n",
       "      <td>115.0</td>\n",
       "      <td>4569922.0</td>\n",
       "      <td>89408.0</td>\n",
       "      <td>61.0</td>\n",
       "      <td>13.96</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>Leptospira interrogans serovar Copenhageni (ba...</td>\n",
       "      <td>Oswaldo Cruz Foundation</td>\n",
       "      <td>Scaffold</td>\n",
       "      <td>PRJNA560059</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>649</th>\n",
       "      <td>SAMN05421689</td>\n",
       "      <td>317.0000</td>\n",
       "      <td>57.0</td>\n",
       "      <td>4594670.0</td>\n",
       "      <td>133191.0</td>\n",
       "      <td>75.0</td>\n",
       "      <td>101.92</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>Leptospira interrogans (bacteria)</td>\n",
       "      <td>DOE - JOINT GENOME INSTITUTE</td>\n",
       "      <td>Scaffold</td>\n",
       "      <td>PRJEB18875</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>650</th>\n",
       "      <td>SAMEA104369440</td>\n",
       "      <td>300.0000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4614703.0</td>\n",
       "      <td>4614703.0</td>\n",
       "      <td>189.0</td>\n",
       "      <td>196.61</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>Leptospira interrogans serovar Manilae (bacteria)</td>\n",
       "      <td>INSTITUT PASTEUR</td>\n",
       "      <td>Chromosome</td>\n",
       "      <td>PRJEB23342</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>516 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      biosample_acc  coverage  # contigs  Total length        N50  \\\n",
       "0      SAMN06434252   30.0000        2.0     4570695.0  4218946.0   \n",
       "1      SAMN06434273   30.0000        2.0     4560160.0  4208129.0   \n",
       "21     SAMN02603847    0.0000        2.0     4627366.0  4277185.0   \n",
       "22     SAMN02603615    0.0000        2.0     3931782.0  3614446.0   \n",
       "23     SAMN02603616    0.0000        2.0     3876235.0  3576473.0   \n",
       "24     SAMN02603617    0.0000        3.0     3956089.0  3603977.0   \n",
       "25     SAMN02603347    0.0000        3.0     3951448.0  3599677.0   \n",
       "26     SAMN02603127    0.0000        2.0     4698134.0  4338762.0   \n",
       "34     SAMN00255265   34.4000       16.0     4705262.0   617754.0   \n",
       "35     SAMN02436336   51.0000      229.0     4679663.0    37187.0   \n",
       "37     SAMN02436629   53.6000       43.0     4741083.0   250073.0   \n",
       "38     SAMN02436466   53.6000       84.0     4813972.0   127002.0   \n",
       "39     SAMN02436545   52.8000       35.0     3914343.0   399563.0   \n",
       "41     SAMN02436335   63.7000       84.0     4370034.0   148695.0   \n",
       "42     SAMN02436628   57.0000       71.0     4918776.0   130328.0   \n",
       "43     SAMN02436547   62.2000       42.0     4580695.0   198881.0   \n",
       "44     SAMN02436546   62.0000       40.0     4660976.0   198307.0   \n",
       "50     SAMN00254365   30.1100      252.0     4545360.0    30507.0   \n",
       "51     SAMN00254363   39.6900      242.0     4545399.0    33096.0   \n",
       "57     SAMN00254358   46.6800      232.0     4548800.0    36256.0   \n",
       "63     SAMN00254352   47.3200      280.0     4544687.0    27958.0   \n",
       "73     SAMN00254342   49.1600      236.0     4547942.0    37545.0   \n",
       "79     SAMN00254336   65.2700      236.0     4550206.0    35609.0   \n",
       "80     SAMN00254335   74.6100      235.0     4547515.0    32986.0   \n",
       "81     SAMN00254334   72.4500      233.0     4549651.0    37915.0   \n",
       "82     SAMN00254333   86.4700      226.0     4546512.0    36023.0   \n",
       "83     SAMN00254332   83.0500      222.0     4545322.0    35769.0   \n",
       "85     SAMN00254330  104.1200      234.0     4545200.0    38517.0   \n",
       "86     SAMN00254329   98.7000      224.0     4548549.0    34635.0   \n",
       "87     SAMN00254328   92.6100      244.0     4549345.0    32537.0   \n",
       "..              ...       ...        ...           ...        ...   \n",
       "620    SAMN10316370   50.0000      205.0     4014286.0    60816.0   \n",
       "621    SAMN10316368   50.0000       28.0     4118320.0   585855.0   \n",
       "622    SAMN10316367   50.0000       18.0     4027080.0  2180649.0   \n",
       "623    SAMN10316389   50.0000       13.0     3849801.0   884966.0   \n",
       "624    SAMN10316379   50.0000       56.0     4156836.0  1004763.0   \n",
       "625    SAMN10316374   50.0000        8.0     3889488.0  2751354.0   \n",
       "626    SAMN10316364   50.0000       78.0     4086283.0   195150.0   \n",
       "627    SAMN10316365   50.0000       63.0     4207523.0   266913.0   \n",
       "628    SAMN10316373   50.0000        7.0     3935107.0  2854332.0   \n",
       "629    SAMN10316443   50.0000       18.0     3746420.0   325090.0   \n",
       "630    SAMN11289419   30.0000        3.0     4740090.0  4306144.0   \n",
       "631    SAMN11289420   30.0000        2.0     4651137.0  4293951.0   \n",
       "632    SAMN11289421   30.0000        4.0     5071735.0  4497709.0   \n",
       "634    SAMN11774879   95.0000      196.0     3706445.0    35480.0   \n",
       "635    SAMN11774878   96.0000      194.0     3728833.0    35794.0   \n",
       "636    SAMN11774876   98.0000      213.0     4530054.0    41872.0   \n",
       "637    SAMN11774877   98.0000      202.0     4525529.0    49707.0   \n",
       "638    SAMN11774874   98.0000      226.0     4522106.0    39552.0   \n",
       "639    SAMN11774875   98.0000      211.0     4529013.0    41894.0   \n",
       "640    SAMN11936725   38.0000        3.0     4361439.0  3978120.0   \n",
       "641    SAMN11936708   30.0000        3.0     4381393.0  3965053.0   \n",
       "642    SAMN12588099  215.0000        3.0     4742398.0  4355568.0   \n",
       "643    SAMN08513768   31.5296       16.0     5336364.0  3284430.0   \n",
       "644    SAMN08596279  498.6260        7.0     5040606.0  4266484.0   \n",
       "645    SAMN12567456  888.0000      112.0     4565779.0    83218.0   \n",
       "646    SAMN12871420  100.0000        4.0     4758488.0  4259066.0   \n",
       "647    SAMN12871448  100.0000        4.0     4755342.0  4255595.0   \n",
       "648    SAMN12567526  152.0000      115.0     4569922.0    89408.0   \n",
       "649    SAMN05421689  317.0000       57.0     4594670.0   133191.0   \n",
       "650  SAMEA104369440  300.0000        1.0     4614703.0  4614703.0   \n",
       "\n",
       "     # misassemblies  # N's per 100 kbp                    species  \\\n",
       "0              128.0             207.89     Leptospira interrogans   \n",
       "1              130.0             612.08     Leptospira interrogans   \n",
       "21              98.0               0.00     Leptospira interrogans   \n",
       "22               0.0               0.00  Leptospira borgpetersenii   \n",
       "23               0.0               0.00  Leptospira borgpetersenii   \n",
       "24               0.0               0.00         Leptospira biflexa   \n",
       "25               0.0               0.00         Leptospira biflexa   \n",
       "26               0.0               0.00     Leptospira interrogans   \n",
       "34              96.0               0.00     Leptospira interrogans   \n",
       "35              85.0               0.00     Leptospira interrogans   \n",
       "37             136.0               0.04     Leptospira interrogans   \n",
       "38               0.0               0.02        Leptospira noguchii   \n",
       "39               0.0               0.05      Leptospira santarosai   \n",
       "41               0.0               0.00          Leptospira weilii   \n",
       "42             150.0               0.00     Leptospira interrogans   \n",
       "43             153.0               0.07     Leptospira interrogans   \n",
       "44             122.0               0.00     Leptospira interrogans   \n",
       "50              54.0              16.17     Leptospira interrogans   \n",
       "51              48.0              19.76     Leptospira interrogans   \n",
       "57              49.0              13.65     Leptospira interrogans   \n",
       "63              39.0              25.00     Leptospira interrogans   \n",
       "73              47.0              16.40     Leptospira interrogans   \n",
       "79              46.0              17.74     Leptospira interrogans   \n",
       "80              53.0              14.69     Leptospira interrogans   \n",
       "81              44.0              17.41     Leptospira interrogans   \n",
       "82              45.0              15.15     Leptospira interrogans   \n",
       "83              53.0              13.16     Leptospira interrogans   \n",
       "85              44.0              15.05     Leptospira interrogans   \n",
       "86              46.0              19.39     Leptospira interrogans   \n",
       "87              47.0              16.05     Leptospira interrogans   \n",
       "..               ...                ...                        ...   \n",
       "620              0.0               0.00     Leptospira mayottensis   \n",
       "621              0.0               0.00  Leptospira dzoumogneensis   \n",
       "622              0.0               0.00  Leptospira bandrabouensis   \n",
       "623              0.0               0.00        Leptospira levettii   \n",
       "624              0.0               0.00          Leptospira meyeri   \n",
       "625              0.0               0.00        Leptospira levettii   \n",
       "626              0.0               0.00          Leptospira idonii   \n",
       "627              0.0               0.00   Leptospira ilyithenensis   \n",
       "628              0.0               0.00        Leptospira levettii   \n",
       "629              0.0               0.00        Leptospira fluminis   \n",
       "630            162.0               0.00     Leptospira interrogans   \n",
       "631            131.0               0.00     Leptospira interrogans   \n",
       "632            155.0               0.00     Leptospira interrogans   \n",
       "634              0.0               0.00  Leptospira borgpetersenii   \n",
       "635              0.0               0.00  Leptospira borgpetersenii   \n",
       "636             84.0               0.00     Leptospira interrogans   \n",
       "637             77.0               0.00     Leptospira interrogans   \n",
       "638             84.0               0.00     Leptospira interrogans   \n",
       "639             83.0               0.00     Leptospira interrogans   \n",
       "640              0.0               0.00          Leptospira weilii   \n",
       "641              0.0               0.00          Leptospira weilii   \n",
       "642            151.0               0.00     Leptospira interrogans   \n",
       "643            149.0               0.00     Leptospira interrogans   \n",
       "644            174.0               0.00     Leptospira interrogans   \n",
       "645             64.0              23.70     Leptospira interrogans   \n",
       "646            165.0               0.00     Leptospira interrogans   \n",
       "647            166.0               0.00     Leptospira interrogans   \n",
       "648             61.0              13.96     Leptospira interrogans   \n",
       "649             75.0             101.92     Leptospira interrogans   \n",
       "650            189.0             196.61     Leptospira interrogans   \n",
       "\n",
       "                                              organism  \\\n",
       "0    Leptospira interrogans serovar Canicola (bacte...   \n",
       "1    Leptospira interrogans serovar Canicola (bacte...   \n",
       "21   Leptospira interrogans serovar Copenhageni str...   \n",
       "22   Leptospira borgpetersenii serovar Hardjo-bovis...   \n",
       "23   Leptospira borgpetersenii serovar Hardjo-bovis...   \n",
       "24   Leptospira biflexa serovar Patoc strain 'Patoc...   \n",
       "25   Leptospira biflexa serovar Patoc strain 'Patoc...   \n",
       "26   Leptospira interrogans serovar Lai str. 56601 ...   \n",
       "34   Leptospira interrogans serovar Canicola str. L...   \n",
       "35   Leptospira interrogans serovar Djasiman str. L...   \n",
       "37       Leptospira interrogans str. C10069 (bacteria)   \n",
       "38      Leptospira noguchii str. 2006001870 (bacteria)   \n",
       "39    Leptospira santarosai str. 2000027870 (bacteria)   \n",
       "41        Leptospira weilii str. 2006001853 (bacteria)   \n",
       "42   Leptospira interrogans str. 2002000621 (bacteria)   \n",
       "43   Leptospira interrogans serovar Pomona str. Pom...   \n",
       "44   Leptospira interrogans serovar Bulgarica str. ...   \n",
       "50   Leptospira interrogans serovar Copenhageni str...   \n",
       "51   Leptospira interrogans serovar Copenhageni str...   \n",
       "57   Leptospira interrogans serovar Copenhageni str...   \n",
       "63   Leptospira interrogans serovar Copenhageni str...   \n",
       "73   Leptospira interrogans serovar Copenhageni str...   \n",
       "79   Leptospira interrogans serovar Copenhageni str...   \n",
       "80   Leptospira interrogans serovar Copenhageni str...   \n",
       "81   Leptospira interrogans serovar Copenhageni str...   \n",
       "82   Leptospira interrogans serovar Copenhageni str...   \n",
       "83   Leptospira interrogans serovar Copenhageni str...   \n",
       "85   Leptospira interrogans serovar Copenhageni str...   \n",
       "86   Leptospira interrogans serovar Copenhageni str...   \n",
       "87   Leptospira interrogans serovar Copenhageni str...   \n",
       "..                                                 ...   \n",
       "620                  Leptospira mayottensis (bacteria)   \n",
       "621               Leptospira dzoumogneensis (bacteria)   \n",
       "622               Leptospira bandrabouensis (bacteria)   \n",
       "623                     Leptospira levettii (bacteria)   \n",
       "624                       Leptospira meyeri (bacteria)   \n",
       "625                     Leptospira levettii (bacteria)   \n",
       "626                       Leptospira idonii (bacteria)   \n",
       "627                Leptospira ilyithenensis (bacteria)   \n",
       "628                     Leptospira levettii (bacteria)   \n",
       "629                     Leptospira fluminis (bacteria)   \n",
       "630                  Leptospira interrogans (bacteria)   \n",
       "631                  Leptospira interrogans (bacteria)   \n",
       "632                  Leptospira interrogans (bacteria)   \n",
       "634               Leptospira borgpetersenii (bacteria)   \n",
       "635               Leptospira borgpetersenii (bacteria)   \n",
       "636                  Leptospira interrogans (bacteria)   \n",
       "637                  Leptospira interrogans (bacteria)   \n",
       "638                  Leptospira interrogans (bacteria)   \n",
       "639                  Leptospira interrogans (bacteria)   \n",
       "640                       Leptospira weilii (bacteria)   \n",
       "641                       Leptospira weilii (bacteria)   \n",
       "642   Leptospira interrogans serovar Hardjo (bacteria)   \n",
       "643  Leptospira interrogans serovar Geyaweera (bact...   \n",
       "644  Leptospira interrogans serovar Weerasinghe (ba...   \n",
       "645  Leptospira interrogans serovar Copenhageni (ba...   \n",
       "646  Leptospira interrogans serovar Canicola (bacte...   \n",
       "647  Leptospira interrogans serovar Canicola (bacte...   \n",
       "648  Leptospira interrogans serovar Copenhageni (ba...   \n",
       "649                  Leptospira interrogans (bacteria)   \n",
       "650  Leptospira interrogans serovar Manilae (bacteria)   \n",
       "\n",
       "                                             Institue  Assembly_status  \\\n",
       "0                     Universidade Federal de Pelotas       Chromosome   \n",
       "1                     Universidade Federal de Pelotas       Chromosome   \n",
       "21                Sao Paulo state (Brazil) Consortium  Complete Genome   \n",
       "22                                  Monash University  Complete Genome   \n",
       "23                                  Monash University  Complete Genome   \n",
       "24                                  Monash University  Complete Genome   \n",
       "25                                   Institut Pasteur  Complete Genome   \n",
       "26                     Chinese National HGC, Shanghai  Complete Genome   \n",
       "34                          J. Craig Venter Institute           Contig   \n",
       "35                          J. Craig Venter Institute           Contig   \n",
       "37                          J. Craig Venter Institute           Contig   \n",
       "38                          J. Craig Venter Institute           Contig   \n",
       "39                          J. Craig Venter Institute           Contig   \n",
       "41                          J. Craig Venter Institute           Contig   \n",
       "42                          J. Craig Venter Institute           Contig   \n",
       "43                          J. Craig Venter Institute           Contig   \n",
       "44                          J. Craig Venter Institute           Contig   \n",
       "50                          J. Craig Venter Institute         Scaffold   \n",
       "51                          J. Craig Venter Institute         Scaffold   \n",
       "57                          J. Craig Venter Institute         Scaffold   \n",
       "63                          J. Craig Venter Institute         Scaffold   \n",
       "73                          J. Craig Venter Institute         Scaffold   \n",
       "79                          J. Craig Venter Institute         Scaffold   \n",
       "80                          J. Craig Venter Institute         Scaffold   \n",
       "81                          J. Craig Venter Institute         Scaffold   \n",
       "82                          J. Craig Venter Institute         Scaffold   \n",
       "83                          J. Craig Venter Institute         Scaffold   \n",
       "85                          J. Craig Venter Institute         Scaffold   \n",
       "86                          J. Craig Venter Institute         Scaffold   \n",
       "87                          J. Craig Venter Institute         Scaffold   \n",
       "..                                                ...              ...   \n",
       "620                     INRS-Institut Armand-Frappier           Contig   \n",
       "621                     INRS-Institut Armand-Frappier           Contig   \n",
       "622                     INRS-Institut Armand-Frappier           Contig   \n",
       "623                     INRS-Institut Armand-Frappier           Contig   \n",
       "624                     INRS-Institut Armand-Frappier           Contig   \n",
       "625                     INRS-Institut Armand-Frappier           Contig   \n",
       "626                     INRS-Institut Armand-Frappier           Contig   \n",
       "627                     INRS-Institut Armand-Frappier           Contig   \n",
       "628                     INRS-Institut Armand-Frappier           Contig   \n",
       "629                     INRS-Institut Armand-Frappier           Contig   \n",
       "630                  Rajarata University of Sri lanka  Complete Genome   \n",
       "631                  Rajarata University of Sri lanka  Complete Genome   \n",
       "632                  Rajarata University of Sri lanka  Complete Genome   \n",
       "634                   INRS - Institut Armand-Frappier           Contig   \n",
       "635                   INRS - Institut Armand-Frappier           Contig   \n",
       "636                   INRS - Institut Armand-Frappier           Contig   \n",
       "637                   INRS - Institut Armand-Frappier           Contig   \n",
       "638                   INRS - Institut Armand-Frappier           Contig   \n",
       "639                   INRS - Institut Armand-Frappier           Contig   \n",
       "640                          Chulalongkorn University  Complete Genome   \n",
       "641                          Chulalongkorn University  Complete Genome   \n",
       "642                         Londrina State University  Complete Genome   \n",
       "643                                              JCVI           Contig   \n",
       "644                                              JCVI           Contig   \n",
       "645                           Oswaldo Cruz Foundation         Scaffold   \n",
       "646  Chinese National Human Genome Center at Shanghai  Complete Genome   \n",
       "647  Chinese National Human Genome Center at Shanghai  Complete Genome   \n",
       "648                           Oswaldo Cruz Foundation         Scaffold   \n",
       "649                      DOE - JOINT GENOME INSTITUTE         Scaffold   \n",
       "650                                  INSTITUT PASTEUR       Chromosome   \n",
       "\n",
       "      bioproject  \n",
       "0    PRJNA376496  \n",
       "1    PRJNA376501  \n",
       "21    PRJNA10687  \n",
       "22    PRJNA16146  \n",
       "23    PRJNA16148  \n",
       "24    PRJNA16153  \n",
       "25    PRJNA20133  \n",
       "26      PRJNA293  \n",
       "34    PRJNA65063  \n",
       "35    PRJNA65061  \n",
       "37    PRJNA65055  \n",
       "38    PRJNA65053  \n",
       "39    PRJNA65051  \n",
       "41    PRJNA65047  \n",
       "42    PRJNA65045  \n",
       "43    PRJNA65043  \n",
       "44    PRJNA65041  \n",
       "50    PRJNA63819  \n",
       "51    PRJNA63817  \n",
       "57    PRJNA63805  \n",
       "63    PRJNA63793  \n",
       "73    PRJNA63771  \n",
       "79    PRJNA63759  \n",
       "80    PRJNA63757  \n",
       "81    PRJNA63755  \n",
       "82    PRJNA63753  \n",
       "83    PRJNA63751  \n",
       "85    PRJNA63747  \n",
       "86    PRJNA63745  \n",
       "87    PRJNA63743  \n",
       "..           ...  \n",
       "620  PRJNA480513  \n",
       "621  PRJNA480513  \n",
       "622  PRJNA480513  \n",
       "623  PRJNA480513  \n",
       "624  PRJNA480513  \n",
       "625  PRJNA480513  \n",
       "626  PRJNA480513  \n",
       "627  PRJNA480513  \n",
       "628  PRJNA480513  \n",
       "629  PRJNA480513  \n",
       "630  PRJNA528695  \n",
       "631  PRJNA528695  \n",
       "632  PRJNA528695  \n",
       "634  PRJNA543681  \n",
       "635  PRJNA543681  \n",
       "636  PRJNA543681  \n",
       "637  PRJNA543681  \n",
       "638  PRJNA543681  \n",
       "639  PRJNA543681  \n",
       "640  PRJNA545909  \n",
       "641  PRJNA545906  \n",
       "642  PRJNA560522  \n",
       "643  PRJNA401006  \n",
       "644  PRJNA401006  \n",
       "645  PRJNA560059  \n",
       "646  PRJNA574874  \n",
       "647  PRJNA574874  \n",
       "648  PRJNA560059  \n",
       "649   PRJEB18875  \n",
       "650   PRJEB23342  \n",
       "\n",
       "[516 rows x 12 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "biosamples_cov = biosamples[(biosamples[\"coverage\"] == 0) | \n",
    "                            (biosamples[\"coverage\"] >= 30) & \n",
    "                            (biosamples[\"N50\"] > 10000) &\n",
    "                           (biosamples[\"# contigs\"] <= 500)][[\"biosample_acc\",\"coverage\",\"# contigs\",\"Total length\",\"N50\",\"# misassemblies\",\"# N's per 100 kbp\",\"species\",\"organism\",'Institue',\"Assembly_status\",\"bioproject\"]]\n",
    "\n",
    "# len(biosamples_cov[biosamples_cov[\"coverage\"] < 50])\n",
    "\n",
    "biosamples_cov\n",
    "# N50 = biosamples['N50'].describe()\n",
    "# num_contigs = biosamples['# contigs'].describe()\n",
    "# num_misasm = biosamples['# misassemblies'].describe()\n",
    "\n",
    "# good_N50 = biosamples[biosamples.N50 > biosamples.N50.quantile(.90)]\n",
    "# # good_length = biosamples[biosamples['Total length'] > biosamples['Total length'].quantile(.75)]\n",
    "# # less_contigs = biosamples[biosamples[\"# contigs\"] < biosamples[\"# contigs\"].quantile(.10)]\n",
    "# intersect = set(good_N50['biosample_acc']).intersection(set(less_contigs['biosample_acc']))\n",
    "\n",
    "# picked = pd.DataFrame(intersect)\n",
    "# print(good_N50)\n",
    "# # N50.plot.kde()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "biosamples_cov.to_csv(seq_path +\"/ncbi_assemblies_filtered.csv\") # write all into a file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
